---
title: "Aged and Postnatal RNA Seq Analysis"
output: html_notebook
editor_options: 
  chunk_output_type: console
---

# Initial Setup:
```{r packages}
library(tidyverse)
library(DESeq2)
library(rnaseqGene)
library(pheatmap)
library(ggalt)
library(fgsea)
library(biomaRt)
library(ggVennDiagram)
library(VennDiagram)
library(ggrepel)
setwd("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Aged RNA Seq")

EnsIDReplace2 <- function(input){
  gene_mapping <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/RNASeq/Cite-seq/GSE145491/Ensembl_labels.csv")
  rnanames <- rownames(input)
  id_list <- setNames(gene_mapping$gene_name, gene_mapping$gene_id)
  rnaname <- character(nrow(input))
  for (i in seq_len(nrow(input))) {
    ensembl_id <- rownames(input)[i]
    if (ensembl_id %in% names(id_list)) {
      rnaname[i] <- id_list[[ensembl_id]]
    } else{
      rnaname[i] <- ensembl_id
    }
  }
  rnaname[rnaname == ""] <- NA
  
  rownames(input) <- make.names(rnaname, unique = TRUE)
  rm(gene_mapping, id_list, ensembl_id, i, rnaname)
}

run_gsea <- function(dds, condition1, condition2, gmt_path, output_gsea, output_deg) {
  res1 <- DESeq2::results(dds, 
                       contrast = c("condition", condition1, condition2),
                       independentFiltering = TRUE,
                       alpha = 0.1)
res1 <- subset(res1, padj < 0.1)
rnk <- setNames(res1$log2FoldChange, rownames(res1))
rnk <- rnk[!is.na(rnk)]
gmt <- gmtPathways(gmt_path)
gsea <- fgsea(pathways = gmt, stats = rnk, minSize = 15, maxSize = 500)
gseadf <- as.data.frame(gsea)
list_columns <- sapply(gseadf, is.list)
gseadf[list_columns] <- lapply(gseadf[list_columns], function(col) sapply(col, paste, collapse = ";"))
write.csv(gseadf, output_gsea)
res1_filtered <- res1[complete.cases(res1), ]
write.csv(res1_filtered, output_deg)
}
```

# Aged Dataset

## Load counts matrix into a DESeqDataSet object

```{r Load RNA Counts Data, warning=FALSE, message=FALSE}
age <- read.csv("Postnatal Compiled (Aged).csv")
rownames(age) <- cts$Geneid
age <- age[, -1]
EnsIDReplace2(age)
age <- age[,-13]
write.csv(age, "aged_genename.csv")
```


```{r Load RNA Counts Data, warning=FALSE, message=FALSE}
age <- read.csv("aged_genename.csv", row.names = 1)
```


## Differential Expression analysis

```{r DESeq}
cda <- read.csv("agedcd.csv") #make coldata file with sample names in one column and then condition in second column
dds2 <- DESeqDataSetFromMatrix(countData = age, colData = cda, design = ~condition)
rlog2 <- rlog(dds2, blind=FALSE)
dds <- DESeq(dds)
rescmp <- results(dds) 
```

### Save results as csv

```{r Save Results, eval=FALSE}
write.csv(rescmp, "C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/MPP3 RNA Seq/completeresults.csv")
```

## Generate PCA Plot

```{r PCA Plot}
pca <- plotPCA(rlogpos)
pcapos <- pca + geom_point(size=5) + 
  theme(
    panel.background = element_rect(fill = "white", color = "black"), # White background
    axis.text = element_blank(),
    axis.ticks = element_blank(),
  )

ggsave("pcapos.png", pcapos, dpi = 600)

```
 + 
  scale_color_manual(labels=c("CMP", "GMP", "MEP", "MPP3 ESAM-", "MPP3 ESAM+", "PG", "PM"), values = c("red","blue", "darkgreen", "purple", "orange", "black", "maroon"))
  lims(x=c(-15, 25), y=c(-15, 25))

## GSEA Analysis

### Run GSEA
```{r GSEA Old}
run_gsea(dds,
         condition1 = "O2P", 
         condition2 = "O2N", 
         gmt_path = "reactome_go_gsea.gmt", 
         output_gsea = "o2_gsea.csv", 
         output_deg = "o2_deg.csv"
         )

run_gsea(dds,
         condition1 = "OMP", 
         condition2 = "OMN", 
         gmt_path = "reactome_go_gsea.gmt", 
         output_gsea = "om_gsea.csv", 
         output_deg = "om_deg.csv"
         )

```

### Overlap
```{r aged overlap}
o2 <- read.csv("o2_deg.csv", row.names = 1)
om <- read.csv("om_deg.csv", row.names = 1)
o2pos <- subset(o2, o2$log2FoldChange > 1)
ompos <- subset(om, om$log2FoldChange > 1)
ovpos_o <- Reduce(intersect, list(rownames(o2pos), rownames(ompos)))
write.csv(ovpos_o, "oldpositiveoverlap.csv")
```

# Postnatal Dataset

## Load counts matrix into a DESeqDataSet object

```{r Load RNA Counts Data, warning=FALSE, message=FALSE}
pn <- read.csv("pn_genename.csv", row.names = 1)

cts3 <- cbind(age, pn)
cd3 <- rbind(cda, cdpn)
```

## Differential Expression analysis

```{r DESeq}
cdpn <- read.csv("pncoldata.csv") #make coldata file with sample names in one column and then condition in second column
cd3 <- rbind(cd, cd2)
dds <- DESeqDataSetFromMatrix(countData = pn, colData = cdpn, design = ~condition)
dds3 <- DESeqDataSetFromMatrix(countData = cts3, colData = cd3, design = ~condition)
rlog <- rlog(dds, blind=FALSE)
rlog3 <- rlog(dds3, blind=FALSE)
vst <- vst(dds, blind = FALSE)
dds <- DESeq(dds)
dds3 <- DESeq(dds3)
rescmp <- results(dds) 

rlogpos <- subset(rlog3, rlog3$condition %in% c("A2N", "AH", "AM", "O2P", "OHP", "OMP", "Y2P", "YH", "YMP"))

keep <- colData(rlog3)$condition %in% c("A2N", "AH", "AM", "O2P", "OHP", "OMP", "Y2P", "YH", "YMP")

rlogpos <- rlog3[, keep]


```



### Overlap
```{r postnatal overlap}
y2 <- read.csv("y2_deg.csv", row.names = 1)
ym <- read.csv("ym_deg.csv", row.names = 1)
y2pos <- subset(y2, y2$log2FoldChange > 1)
ympos <- subset(ym, ym$log2FoldChange > 1)
ovpos_y <- Reduce(intersect, list(rownames(y2pos), rownames(ympos)))
write.csv(ovpos_y, "youngpositiveoverlap.csv")
genelist <- list(Y2 = rownames(y2pos), YM = rownames(ympos))

ggVennDiagram(genelist, label_alpha = 0) + 
  scale_fill_gradient(low = "white", high = "lightblue")

y2neg <- subset(y2, y2$log2FoldChange < 0)
y2neglist <- rownames(y2neg)
ymneg <- subset(ym, ym$log2FoldChange < 0)
ymneg <- ymneg[order(ymneg$log2FoldChange),]
ymneglist <- rownames(ymneg)[1:39]
pnneg <- Reduce(intersect, list(rownames(y2neg), rownames(ymneg)))
pnneg <- unique(c(y2neglist, ymneglist))
write.csv(pnneg, "youngnegativedeg.csv")
genelist <- list(Y2 = y2neglist, YM = ymneglist)

ggVennDiagram(genelist, label_alpha = 0) + 




```


### Overall overlap
```{r}
ovpos <- Reduce(intersect, list(rownames(o2pos), rownames(ompos), rownames(y2pos), rownames(ympos)))
write.csv(ovpos, "overallpositiveoverlap.csv")

o2gene <- rownames(o2pos)
omgene <- rownames(ompos)
y2gene <- rownames(y2pos)
ymgene <- rownames(ympos)
genelist <- list(O2 = o2gene, OM = omgene, Y2 = y2gene, YM = ymgene)

ggVennDiagram(genelist, label_alpha = 0) + 
  scale_fill_gradient(low = "white", high = "lightblue")

```


# Convert to fpkm
```{r fpkm}
ensembl <- useMart("ensembl", dataset = "mmusculus_gene_ensembl")
attributes <- c("mgi_symbol", "ensembl_gene_id", "start_position", "end_position")
gene_names <- rownames(dds)
feature_data <- getBM(attributes = attributes, filter = "mgi_symbol", values = gene_names, mart = ensembl)
feature_data$length <- feature_data$end_position - feature_data$start_position + 1

gene_names <- rownames(dds)
unique_gene_names <- make.unique(gene_names)
rownames(dds) <- unique_gene_names

rownames(feature_data) <- make.unique(feature_data$mgi_symbol)
mcols(dds)$basepairs <- feature_data$length


rownames(feature_data) <- feature_data$mgi_symbol
mcols(dds)$basepairs <- feature_data[rownames(dds), "length"]

fpkm <- fpkm(dds)

```

```{r save fpkm}
write.csv(fpkm ,"agedfpkm.csv")
```

```{r Heatmap of Specific Genes}
gfppn <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/RNASeq/Gene Signatures/gfppospnsignature.csv")
gfppn <- gfppn[,-1]
negative <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/RNASeq/Gene Signatures/youngnegativedeg.csv")
negative <- negative[,-1]
genelist <- c(gfppn, negative)

genes <- read.csv("heatmapgenes.csv")

fetal <- read.csv("Fetal Signature.csv")
fetal2 <- read.csv("fetalsignature2.csv")

# Subset normalized counts to include only the genes of interest

conditions <- c("Y2P", "YMP", "Y2N", "YMN", "AM", "A2N")
cdsub <- cd[cd$condition %in% conditions, ]
cdsub$condition <- factor(cdsub$condition, levels = c("YMN", "YMP", "AM", "Y2N", "Y2P", "A2N"))
cdsub <- cdsub[order(cdsub$condition), ]
rlogsub <- rlog[ ,colnames(rlog) %in% cdsub$X]


rloggenes <- assay(rlogsub)
rloggenes <- rloggenes[rownames(rloggenes) %in% fetal2$X, ]
rloggenes <- rloggenes[order(match(rownames(rloggenes), fetal2$X)), ]
rloggenes <- rloggenes[, cdsub$X]




# Define color palette
color_palette <- colorRampPalette(c("blue", "white", "red"))(100)

# Create a heatmap
heatmap <- pheatmap(rloggenes, 
         scale = "row", 
         cluster_rows = FALSE, 
         cluster_cols = FALSE, 
         color = color_palette,
         show_rownames = TRUE, 
         show_colnames = TRUE,
         main = "Heatmap",
         treeheight_row = 0,
         cellwidth = 10,
         cellheight = 10)
heatmap
ggsave("pnheatmapfetal2.png", heatmap, height = 27, width = 8, dpi = 600)

```


```{r HSC dds creation}
agehsc <- age[,(13:14)]
pnhsc <- pn[,(28:33)]
hsc <- cbind(agehsc, pnhsc)

cdahsc <- cda[(13:14),]
cdpnhsc <- cdpn[(28:33),]
cdhsc <- rbind(cdahsc, cdpnhsc)

dds <- DESeqDataSetFromMatrix(countData = hsc, colData = cdhsc, design = ~condition)
rlog2 <- rlog(dds, blind=FALSE)
dds <- DESeq(dds)
rescmp <- results(dds) 

rm(age, agehsc, cd, cda, cdahsc, cdpn, cdpnhsc, cts, genelist, o2, o2pos, om, ompos, pca, pn, pnhsc, rescmp, y2, y2neg, y2pos, ym, ympos, ymneg, negative, o2gene, omgene, overlap, ovpos, ovpos_o, ovpos_y, pnneg, y2gene, y2neglist, ymgene, ymneglist)
```

```{r PCA Plot}
pca <- plotPCA(rlog2, returnData = TRUE)
pca$group <- factor(pca$group, levels = c("AH", "OHP", "YH"), labels = c("Adult HSCs", "Aged GFP+ HSCs", "Young GFP+ HSCs"))
hscpca <- ggplot(pca, aes(PC1, PC2, color = group)) + 
  geom_point(size=5) + 
  theme(
    panel.background = element_rect(fill = "white", color = "black"), # White background
    axis.text = element_blank(),
    axis.ticks = element_blank(),
  ) 

ggsave("hscpca.png", plot = hscpca, dpi = 600)
```

```{r DEGs and GSEA}
run_gsea(dds,
         condition1 = "OHP", 
         condition2 = "YH", 
         gmt_path = "reactome_go_gsea.gmt", 
         output_gsea = "oldvspn_hsc_gsea.csv", 
         output_deg = "oldvspn_hsc_deg.csv"
         )

run_gsea(dds,
         condition1 = "OHP", 
         condition2 = "AH", 
         gmt_path = "reactome_go_gsea.gmt", 
         output_gsea = "oldhsc_gsea.csv", 
         output_deg = "oldhsc_deg.csv"
         )

run_gsea(dds,
         condition1 = "YH", 
         condition2 = "AH", 
         gmt_path = "reactome_go_gsea.gmt", 
         output_gsea = "pnhsc_gsea.csv", 
         output_deg = "pnhsc_deg.csv"
         )
```

```{r Overlap vs AH}
degold <- read.csv("oldhsc_deg.csv", row.names = 1)
degpn <- read.csv("pnhsc_deg.csv", row.names = 1)
posold <- subset(degold, degold$log2FoldChange > 1)
pospn <- subset(degpn, degpn$log2FoldChange > 1)
hscposov <- Reduce(intersect, list(rownames(posold), rownames(pospn)))
write.csv(hscposov, "oldvspn_posoverlap.csv")
genelist <- list(old = rownames(posold), pn = rownames(pospn))

posvenn <- ggVennDiagram(genelist, label_alpha = 0) + 
  scale_fill_gradient(low = "white", high = "lightblue")
ggsave("posvenn.png", posvenn, dpi = 600)

negold <- subset(degold, degold$log2FoldChange < -1)
negpn <- subset(degpn, degpn$log2FoldChange < -1)
hscnegov <- Reduce(intersect, list(rownames(negold), rownames(negpn)))
write.csv(hscnegov, "oldvspn_negoverlap.csv")
genelist <- list(old = rownames(negold), pn = rownames(negpn))

negvenn <- ggVennDiagram(genelist, label_alpha = 0) + 
  scale_fill_gradient(low = "white", high = "lightblue")
ggsave("negvenn.png", negvenn, dpi = 600)

```

```{r Overlapv2}
degold <- read.csv("oldhsc_deg.csv", row.names = 1)
degv <- read.csv("oldvspn_hsc_deg.csv", row.names = 1)
posold <- subset(degold, degold$log2FoldChange > 1)
posv <- subset(degv, degv$log2FoldChange > 1)
hscoldovpos <- Reduce(intersect, list(rownames(posold), rownames(posv)))
write.csv(hscoldovpos, "old_posoverlap.csv")
genelist <- list(vsAdult = rownames(posold), vsPN = rownames(posv))

oldposvenn <- ggVennDiagram(genelist, label_alpha = 0) + 
  scale_fill_gradient(low = "white", high = "lightblue")
ggsave("oldposvenn.png", oldposvenn, dpi = 600, width = 10, height = 10)

negold <- subset(degold, degold$log2FoldChange > -1)
negv <- subset(degv, degv$log2FoldChange > -1)
hscoldovneg <- Reduce(intersect, list(rownames(negold), rownames(negv)))
write.csv(hscoldovneg, "old_negoverlap.csv")
genelist <- list(vsAdult = rownames(negold), vsPN = rownames(negv))

oldnegvenn <- ggVennDiagram(genelist, label_alpha = 0) + 
  scale_fill_gradient(low = "white", high = "lightblue")
ggsave("oldnegvenn.png", oldnegvenn, dpi = 600, width = 10, height = 10)

```

```{r 6-24-25}
run_gsea(dds,
         condition1 = "Y2P", 
         condition2 = "Y2N", 
         gmt_path = "m3.gtrd.v2025.1.Mm.symbols.gmt", 
         output_gsea = "mpp2tf_gsea.csv", 
         output_deg = "mpp2tf_deg.csv"
         )

run_gsea(dds,
         condition1 = "Y2P", 
         condition2 = "Y2N", 
         gmt_path = "m1.all.v2025.1.Mm.symbols.gmt", 
         output_gsea = "mpp2chr_gsea.csv", 
         output_deg = "mpp2chr_deg.csv"
         )

run_gsea(dds,
         condition1 = "Y1P", 
         condition2 = "Y1N", 
         gmt_path = "m3.gtrd.v2025.1.Mm.symbols.gmt", 
         output_gsea = "mpp1tf_gsea.csv", 
         output_deg = "mpp1tf_deg.csv"
         )

run_gsea(dds,
         condition1 = "Y1P", 
         condition2 = "Y1N", 
         gmt_path = "m1.all.v2025.1.Mm.symbols.gmt", 
         output_gsea = "mpp1chr_gsea.csv", 
         output_deg = "mpp1chr_deg.csv"
         )

run_gsea(dds,
         condition1 = "YMP", 
         condition2 = "YMN", 
         gmt_path = "m3.gtrd.v2025.1.Mm.symbols.gmt", 
         output_gsea = "mkptf_gsea.csv", 
         output_deg = "mkptf_deg.csv"
         )

run_gsea(dds,
         condition1 = "YMP", 
         condition2 = "YMN", 
         gmt_path = "m1.all.v2025.1.Mm.symbols.gmt", 
         output_gsea = "mkpchr_gsea.csv", 
         output_deg = "mkpchr_deg.csv"
         )
```

```{r}
# Transpose the rlog matrix to get samples as rows
rlog_df <- as.data.frame(t(assay(rlog3)))

# Build the plot data
plot_df <- data.frame(
  Ctnnal1 = rlog_df$Ctnnal1,
  Hhex = rlog_df$Hhex,
  Sample = rownames(rlog_df)  # assumes sample names are stored here
)

ggplot(plot_df, aes(x = Ctnnal1, y = Hhex, label = Sample)) +
  geom_point(color = "#0072B2", size = 2, alpha = 0.7) +
  geom_text_repel(size = 3) +
  labs(
    title = "rlog Expression: Ctnnal1 vs Hhex (Labeled by Sample)",
    x = "Ctnnal1 (rlog)",
    y = "Hhex (rlog)"
  ) +
  theme_minimal()


cor(plot_df$Ctnnal1, plot_df$Hhex, method = "pearson")
```

```{r}
# Transpose so samples are rows
raw_df <- as.data.frame(t(cts3))

# Pull the genes of interest
plot_df <- data.frame(
  Ctnnal1 = raw_df$Ctnnal1,
  Bex4 = raw_df$Bex4,
  Sample = rownames(raw_df)
)

# Plot with raw counts (optionally log-transform for better spread)
ggplot(plot_df, aes(x = Ctnnal1, y = Bex4, label = Sample)) +
  geom_point(color = "#0072B2", size = 2, alpha = 0.7) +
  geom_text(vjust = -0.6, size = 3) +
  labs(
    title = "Raw Counts: Ctnnal1 vs Bex4",
    x = "Ctnnal1 (raw counts)",
    y = "Bex4 (raw counts)"
  ) +
  theme_minimal()

cor(plot_df$Ctnnal1, plot_df$Hmga2, method = "pearson")

```

```{r}
# Assume 'expr_matrix' is your normalized gene expression matrix:
# rows = genes, columns = samples

# Transpose for correlation: genes as rows, samples as columns
expr_mat <- as.data.frame(t(cts3))

# Sanity check
if (!"Ctnnal1" %in% colnames(expr_mat)) stop("Ctnnal1 not found in dataset")

# Compute correlation of all genes with Ctnnal1
cor_vec <- apply(expr_mat, 2, function(x) cor(x, expr_mat$Ctnnal1, method = "pearson"))

# Remove Ctnnal1 itself from results
cor_vec <- cor_vec[names(cor_vec) != "Ctnnal1"]

# Sort and view top positively correlated genes
top_pos <- sort(cor_vec, decreasing = TRUE)[1:50]

# Sort and view top negatively correlated genes
top_neg <- sort(cor_vec, decreasing = FALSE)[1:50]

# Combine and view
top_correlated <- data.frame(
  Gene = c(names(top_pos), names(top_neg)),
  Correlation = c(top_pos, top_neg)
)

barplot(top_pos, las = 2, col = "steelblue", main = "Top Correlated Genes")
barplot(top_neg, las = 2, col = "steelblue", main = "Lowest Correlated Genes")

write.csv(top_correlated, "ctnnal1_cor_allsamples.csv")

# Convert to data frame for ggplot
top_df <- data.frame(
  Gene = factor(names(top_pos), levels = names(top_pos)),
  Correlation = top_pos
)

# Plot
ggplot(top_df, aes(x = Gene, y = Correlation)) +
  geom_bar(stat = "identity", fill = "steelblue") +
  theme_minimal() +
  labs(title = "Top Correlated Genes", x = "", y = "Pearson Correlation") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Convert to data frame for ggplot
top_df <- data.frame(
  Gene = factor(names(top_neg), levels = names(top_neg)),
  Correlation = top_neg
)

# Plot
ggplot(top_df, aes(x = Gene, y = Correlation)) +
  geom_bar(stat = "identity", fill = "steelblue") +
  theme_minimal() +
  labs(title = "Lowest Correlated Genes", x = "", y = "Pearson Correlation") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))


```

```{r 8/22/2025}
# ---- Load metadata ----
coldata_raw <- read.csv("combinedcoldata.csv")

# ---- Parse sample names into tidy metadata ----
coldata <- coldata_raw %>%
  rowwise() %>%
  mutate(
    # Age
    age = case_when(
      grepl("^Old", X) ~ "Old",
      grepl("^Y", X) ~ "Postnatal",
      grepl("^A", X) ~ "Adult",
      TRUE ~ NA_character_
    ),
    
    # Cell type
    celltype = case_when(
      grepl("HSC", X, ignore.case = TRUE, fixed=TRUE) ~ "HSC",
      grepl("MPP1", X, ignore.case = TRUE, fixed=TRUE) ~ "MPP1",
      grepl("MPP2", X, ignore.case = TRUE, fixed=TRUE) ~ "MPP2",
      grepl("MKP", X, ignore.case = TRUE, fixed=TRUE) ~ "MkP",
      grepl("H", X) ~ "HSC",
      grepl("1", X) ~ "MPP1",
      grepl("2", X) ~ "MPP2",
      grepl("M", X, ignore.case = TRUE) ~ "MkP",
      TRUE ~ NA_character_
    ),
    
    # GFP
    GFP = case_when(
      # Old samples
      grepl("POS", X, ignore.case = TRUE) ~ "POS",
      grepl("NEG", X, ignore.case = TRUE) ~ "NEG",
      # Y samples
      grepl("^Y.*P", X) ~ "POS",
      grepl("^Y.*N", X) ~ "NEG",
      grepl("YH", X) ~ "POS",
      # A samples â†’ all NEG
      grepl("^A", X) ~ "NEG",
      TRUE ~ NA_character_
    ),
    
    # Replicate
    replicate = sub(".*[_-](\\d+)$", "\\1", X)
  ) %>%
  ungroup()

# Keep only complete metadata
coldata <- coldata %>% filter(!is.na(age) & !is.na(celltype) & !is.na(GFP))

# Standardize sample names to match counts file (replace "-" with "." for old samples)
rownames(coldata) <- gsub("-", ".", coldata$X)

# ---- Load counts ----
counts <- read.csv("combinedcounts.csv", row.names = 1)

# Align counts and metadata
common_samples <- intersect(rownames(coldata), colnames(counts))
coldata <- coldata[common_samples, ]
counts <- counts[, common_samples]

# ---- Build DESeq2 dataset ----
dds <- DESeqDataSetFromMatrix(
  countData = counts,
  colData = coldata,
  design = ~ age + celltype + GFP
)

# Prefilter low counts
dds <- dds[rowSums(counts(dds)) > 10, ]

# Run DESeq2
dds <- DESeq(dds)

# ---- PCA ----
vsd <- vst(dds)
pcaData <- plotPCA(vsd, intgroup = c("age","celltype","GFP"), returnData = TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))

# Save PCA as object
pca <- ggplot(pcaData, aes(PC1, PC2, color=age, shape=celltype, alpha=ifelse(celltype == "HSC", 1, 0.4))) +
  geom_point(size=3) +
  xlab(paste0("PC1: ", percentVar[1], "% variance")) +
  ylab(paste0("PC2: ", percentVar[2], "% variance")) +
  theme_bw() + 
  guides(alpha = "none")
pca

# ---- Heatmap of top variable genes ----
# Get variance per gene
geneVars <- rowVars(assay(vsd))

# Drop NAs
geneVars <- geneVars[!is.na(geneVars)]

# Select top 500 variable genes (or fewer if not enough)
topVarGenes <- head(order(geneVars, decreasing = TRUE), 500)

# Check sample alignment before plotting
stopifnot(all(colnames(assay(vsd)) == rownames(coldata)))

# Make heatmap
heatmap <- pheatmap(
  assay(vsd)[topVarGenes, ],
  cluster_rows = TRUE,
  show_rownames = FALSE,
  cluster_cols = TRUE,
  annotation_col = coldata,
  silent = TRUE
)

# ---- Key contrasts for HSC GFP+ ----
# Subset to HSC & GFP+
hscsamples <- rownames(coldata[coldata$celltype == "HSC", ])

dds_hsc_gfp_pos <- dds[, hscsamples]

dds_hsc_gfp_pos <- DESeq(dds_hsc_gfp_pos)

# Now run contrast
res_postnatal_vs_adult <- results(
  dds_hsc_gfp_pos,
  contrast = c("age", "Postnatal", "Adult")
)
res_old_vs_adult <- results(dds_hsc_gfp_pos,
    contrast = c("age","Old","Adult")
)

res_old_vs_postnatal <- results(dds_hsc_gfp_pos,
    contrast = c("age","Old","Postnatal"),
)

# ---- Save results ----
write.csv(as.data.frame(res_postnatal_vs_adult), "res_postnatal_vs_adult.csv")
write.csv(as.data.frame(res_old_vs_adult), "res_old_vs_adult.csv")
write.csv(as.data.frame(res_old_vs_postnatal), "res_old_vs_postnatal.csv")

```


