---
title: "Aged and Postnatal RNA Seq Analysis"
output: html_notebook
editor_options: 
  chunk_output_type: console
---

# Initial Setup:
```{r packages}
library(tidyverse)
library(DESeq2)
library(rnaseqGene)
library(pheatmap)
library(ggalt)
library(fgsea)
library(biomaRt)
library(ggVennDiagram)
library(VennDiagram)
setwd("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Aged RNA Seq")

EnsIDReplace2 <- function(input){
  gene_mapping <- read.csv("C:/Users/rohit/OneDrive - Loyola University Chicago/Zhang Lab/RNASeq/Cite-seq/GSE145491/Ensembl_labels.csv")
  rnanames <- rownames(input)
  id_list <- setNames(gene_mapping$gene_name, gene_mapping$gene_id)
  rnaname <- character(nrow(input))
  for (i in seq_len(nrow(input))) {
    ensembl_id <- rownames(input)[i]
    if (ensembl_id %in% names(id_list)) {
      rnaname[i] <- id_list[[ensembl_id]]
    } else{
      rnaname[i] <- ensembl_id
    }
  }
  rnaname[rnaname == ""] <- NA
  
  rownames(input) <- make.names(rnaname, unique = TRUE)
  rm(gene_mapping, id_list, ensembl_id, i, rnaname)
}

run_gsea <- function(dds, condition1, condition2, gmt_path, output_gsea, output_deg) {
  res1 <- DESeq2::results(dds, 
                       contrast = c("condition", condition1, condition2),
                       independentFiltering = TRUE,
                       alpha = 0.1)
res1 <- subset(res1, padj < 0.1)
rnk <- setNames(res1$log2FoldChange, rownames(res1))
rnk <- rnk[!is.na(rnk)]
gmt <- gmtPathways(gmt_path)
gsea <- fgsea(pathways = gmt, stats = rnk, minSize = 15, maxSize = 500)
gseadf <- as.data.frame(gsea)
list_columns <- sapply(gseadf, is.list)
gseadf[list_columns] <- lapply(gseadf[list_columns], function(col) sapply(col, paste, collapse = ";"))
write.csv(gseadf, output_gsea)
res1_filtered <- res1[complete.cases(res1), ]
write.csv(res1_filtered, output_deg)
}
```

# Aged Dataset

## Load counts matrix into a DESeqDataSet object

```{r Load RNA Counts Data, warning=FALSE, message=FALSE}
cts <- read.csv("Postnatal Compiled (Aged).csv")
rownames(cts) <- cts$Geneid
cts <- cts[, -1]
EnsIDReplace2(cts)
cts <- cts[,-13]
write.csv(cts, "aged_genename.csv")
```


```{r Load RNA Counts Data, warning=FALSE, message=FALSE}
cts2 <- read.csv("aged_genename.csv", row.names = 1)
```


## Differential Expression analysis

```{r DESeq}
cd2 <- read.csv("agedcd.csv") #make coldata file with sample names in one column and then condition in second column
dds2 <- DESeqDataSetFromMatrix(countData = cts2, colData = cd2, design = ~condition)
rlog2 <- rlog(dds2, blind=FALSE)
dds <- DESeq(dds)
rescmp <- results(dds) 
```

### Save results as csv

```{r Save Results, eval=FALSE}
write.csv(rescmp, "C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/MPP3 RNA Seq/completeresults.csv")
```

## Generate PCA Plot

```{r PCA Plot}
pca <- plotPCA(rlog)
pca + geom_point(size=5) + 
  theme(
    panel.background = element_rect(fill = "white", color = "black"), # White background
    axis.text = element_blank(),
    axis.ticks = element_blank(),
  )

```
 + 
  scale_color_manual(labels=c("CMP", "GMP", "MEP", "MPP3 ESAM-", "MPP3 ESAM+", "PG", "PM"), values = c("red","blue", "darkgreen", "purple", "orange", "black", "maroon"))
  lims(x=c(-15, 25), y=c(-15, 25))

## GSEA Analysis

### Run GSEA
```{r GSEA Old}
run_gsea(dds,
         condition1 = "O2P", 
         condition2 = "O2N", 
         gmt_path = "reactome_go_gsea.gmt", 
         output_gsea = "o2_gsea.csv", 
         output_deg = "o2_deg.csv"
         )

run_gsea(dds,
         condition1 = "OMP", 
         condition2 = "OMN", 
         gmt_path = "reactome_go_gsea.gmt", 
         output_gsea = "om_gsea.csv", 
         output_deg = "om_deg.csv"
         )

```

### Overlap
```{r aged overlap}
o2 <- read.csv("o2_deg.csv", row.names = 1)
om <- read.csv("om_deg.csv", row.names = 1)
o2pos <- subset(o2, o2$log2FoldChange > 1)
ompos <- subset(om, om$log2FoldChange > 1)
ovpos_o <- Reduce(intersect, list(rownames(o2pos), rownames(ompos)))
write.csv(ovpos_o, "oldpositiveoverlap.csv")
```

# Postnatal Dataset

## Load counts matrix into a DESeqDataSet object

```{r Load RNA Counts Data, warning=FALSE, message=FALSE}
cts <- read.csv("pn_genename.csv", row.names = 1)

cts3 <- cbind(cts, cts2)
```

## Differential Expression analysis

```{r DESeq}
cd <- read.csv("pncoldata.csv") #make coldata file with sample names in one column and then condition in second column
cd3 <- rbind(cd, cd2)
dds <- DESeqDataSetFromMatrix(countData = cts, colData = cd, design = ~condition)
dds3 <- DESeqDataSetFromMatrix(countData = cts3, colData = cd3, design = ~condition)
rlog <- rlog(dds, blind=FALSE)
rlog3 <- rlog(dds3, blind=FALSE)
vst <- vst(dds, blind = FALSE)
dds <- DESeq(dds)
rescmp <- results(dds) 
```



### Overlap
```{r postnatal overlap}
y2 <- read.csv("y2_deg.csv", row.names = 1)
ym <- read.csv("ym_deg.csv", row.names = 1)
y2pos <- subset(y2, y2$log2FoldChange > 1)
ympos <- subset(ym, ym$log2FoldChange > 1)
ovpos_y <- Reduce(intersect, list(rownames(y2pos), rownames(ympos)))
write.csv(ovpos_y, "youngpositiveoverlap.csv")

y2neg <- subset(y2, y2$log2FoldChange < 0)
y2neglist <- rownames(y2neg)
ymneg <- subset(ym, ym$log2FoldChange < 0)
ymneg <- ymneg[order(ymneg$log2FoldChange),]
ymneglist <- rownames(ymneg)[1:39]
pnneg <- Reduce(intersect, list(rownames(y2neg), rownames(ymneg)))
pnneg <- unique(c(y2neglist, ymneglist))
write.csv(pnneg, "youngnegativedeg.csv")

```


### Overall overlap
```{r}
ovpos <- Reduce(intersect, list(rownames(o2pos), rownames(ompos), rownames(y2pos), rownames(ympos)))
write.csv(ovpos, "overallpositiveoverlap.csv")

o2gene <- rownames(o2pos)
omgene <- rownames(ompos)
y2gene <- rownames(y2pos)
ymgene <- rownames(ympos)
genelist <- list(O2 = o2gene, OM = omgene, Y2 = y2gene, YM = ymgene)

ggVennDiagram(genelist, label_alpha = 0) + 
  scale_fill_gradient(low = "white", high = "lightblue")

```


# Convert to fpkm
```{r fpkm}
ensembl <- useMart("ensembl", dataset = "mmusculus_gene_ensembl")
attributes <- c("mgi_symbol", "ensembl_gene_id", "start_position", "end_position")
gene_names <- rownames(dds)
feature_data <- getBM(attributes = attributes, filter = "mgi_symbol", values = gene_names, mart = ensembl)
feature_data$length <- feature_data$end_position - feature_data$start_position + 1

gene_names <- rownames(dds)
unique_gene_names <- make.unique(gene_names)
rownames(dds) <- unique_gene_names

rownames(feature_data) <- make.unique(feature_data$mgi_symbol)
mcols(dds)$basepairs <- feature_data$length


rownames(feature_data) <- feature_data$mgi_symbol
mcols(dds)$basepairs <- feature_data[rownames(dds), "length"]

fpkm <- fpkm(dds)

```

```{r save fpkm}
write.csv(fpkm ,"agedfpkm.csv")
```

```{r Heatmap of Specific Genes}
gfppn <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/RNASeq/Gene Signatures/gfppospnsignature.csv")
gfppn <- gfppn[,-1]
negative <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/RNASeq/Gene Signatures/youngnegativedeg.csv")
negative <- negative[,-1]
genelist <- c(gfppn, negative)

# Subset normalized counts to include only the genes of interest

conditions <- c("Y2P", "YMP", "Y2N", "YMN", "AM", "A2N")
cdsub <- cd[cd$condition %in% conditions, ]
cdsub$condition <- factor(cdsub$condition, levels = c("Y2P", "Y2N", "A2N", "YMP", "YMN", "AM"))
cdsub <- cdsub[order(cdsub$condition), ]
rlogsub <- rlog[ ,colnames(rlog) %in% cdsub$X]


rloggenes <- assay(rlogsub)
rloggenes <- rloggenes[rownames(rloggenes) %in% genelist, ]
rloggenes <- rloggenes[, cdsub$X]




# Define color palette
color_palette <- colorRampPalette(c("blue", "white", "red"))(100)

# Create a heatmap
heatmap <- pheatmap(rloggenes, 
         scale = "row", 
         cluster_rows = FALSE, 
         cluster_cols = FALSE, 
         color = color_palette,
         show_rownames = TRUE, 
         show_colnames = TRUE,
         main = "Heatmap",
         treeheight_row = 0,
         cellwidth = 10,
         cellheight = 10)
heatmap
ggsave("pnheatmap7.png", heatmap, height = 15, width = 10, dpi = 600)

```
