---
title: "GSE128761"
output: html_notebook
editor_options: 
  chunk_output_type: console
---

# HPC Dataset

```{r libraries, results='hide'}
library(Seurat)
library(ggplot2)
library(patchwork)
library(dplyr)
library(biomaRt)
library(pheatmap)
library(fgsea)
library(patchwork)
library(scales)
library(ggbreak)
library(ggcyto)
library(DESeq2)
options(future.globals.maxSize = 10000 * 1024^2)
setwd("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/RNASeq/HPC GSE128761")

run_gsea <- function(markers, gmt_path, output_gsea) {
rnk <- setNames(markers$avg_log2FC, rownames(markers))
rnk <- rnk[!is.na(rnk)]
gmt <- gmtPathways(gmt_path)
gsea <- fgsea(pathways = gmt, stats = rnk, minSize = 15, maxSize = 500)
gseadf <- as.data.frame(gsea)
list_columns <- sapply(gseadf, is.list)
gseadf[list_columns] <- lapply(gseadf[list_columns], function(col) sapply(col, paste, collapse = ";"))
write.csv(gseadf, output_gsea)
}

EnsIDReplace2 <- function(input, output){
  gene_mapping <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/RNASeq/Cite-seq/GSE145491/Ensembl_labels.csv")
  rna <- input
  rnanames <- rownames(rna)
  id_list <- setNames(gene_mapping$gene_name, gene_mapping$gene_id)
  rnaname <- character(nrow(rna))
  for (i in seq_len(nrow(rna))) {
    ensembl_id <- rownames(rna)[i]
    if (ensembl_id %in% names(id_list)) {
      rnaname[i] <- id_list[[ensembl_id]]
    } else{
      rnaname[i] <- ensembl_id
    }
  }
  rnaname[rnaname == ""] <- NA
  
  rownames(rna) <- rnaname
  output <- rna
  rm(gene_mapping, id_list, ensembl_id, i, rnaname)
}
```


``` {r Load Seurat Object and Cluster}
data <- Read10X(data.dir = "Data/")
so <- CreateSeuratObject(counts = data)
so <- NormalizeData(so)
so <- FindVariableFeatures(so)
so <- ScaleData(so)
so <- RunPCA(so)
so <- FindNeighbors(so, dims = 1:30)
so <- FindClusters(so)
so <- RunUMAP(so, dims = 1:30)
rm(data)
```

```{r}
so$orig.ident <- factor(so$orig.ident, levels = c("HPC.E16.5", "HPC.P7", "HPC.P14", "HPC.Adult"))

dimacat <- DimPlot(so, group.by = "acat", cols = c("grey", "blue"))
dimacat
ggsave("hpcumapacat.png", dimacat, height = 8, width = 10, dpi = 600)

dimplot <- DimPlot(so, group.by = "seurat_clusters")
dimplot
ggsave("hpcumap.png", dimplot, height = 8, width = 10, dpi = 600)

dimtime <- DimPlot(so, group.by = "orig.ident")
dimtime
ggsave("hpcumapage.png", dimtime, height = 8, width = 10, dpi = 600)

```

, group.by = c("orig.ident")

```{r Bar Chart}
Idents(so) <- "orig.ident"
test <- FetchData(so, vars = c("orig.ident", "Ctnnal1"))
test <- data.frame(Cell = rownames(test), Cluster = test$orig.ident, test = test$Ctnnal1)
test <- test[order(test$Cluster),]
test$Cell <- factor(test$Cell, levels = test$Cell)

testplot <- ggplot(test, aes(x = Cell, y = test, fill = as.factor(Cluster))) + 
    geom_bar(stat = "identity", width = 1) +
  theme(
    axis.text = element_blank(), 
    axis.title = element_blank(), 
    axis.ticks = element_blank(), 
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1)
  )
testplot

Idents(so) <- "orig.ident"
test2 <- FetchData(so, vars = c("orig.ident", "seurat_clusters", "Ctnnal1"))
test2 <- data.frame(Cell = rownames(test2), TimePoint = test2$orig.ident, Cluster = test2$seurat_clusters, 
                    Ctnnal1 = test2$Ctnnal1)
test2 <- test2[order(test2$Cluster),]
test2$Cell <- factor(test2$Cell, levels = test2$Cell)

test2p <- ggplot(test2, aes(x=Cell, y=Ctnnal1, fill=as.factor(Cluster))) + 
    geom_bar(stat = "identity", width = 1) +
    facet_wrap(~ TimePoint, scales = "free_x") + 
  theme(
    axis.text.x = element_blank(), 
    axis.title.x = element_blank(), 
    axis.ticks.x = element_blank(), 
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1)
  ) + 
  labs(y="Ctnnal1")
test2p

Idents(so) <- "orig.ident"
test2 <- FetchData(so, vars = c("orig.ident", "seurat_clusters", "Ctnnal1"))
test2 <- data.frame(Cell = rownames(test2), TimePoint = test2$orig.ident, Cluster = test2$seurat_clusters, 
                    Ctnnal1 = test2$Ctnnal1)
test2 <- test2[order(test2$Cluster),]
test2$Cell <- factor(test2$Cell, levels = test2$Cell)

test2$HPC.E16 <- ifelse(test2$TimePoint == "HPC.E16.5", test2$Ctnnal1, 0)
test2$HPC.P7 <- ifelse(test2$TimePoint == "HPC.P7", test2$Ctnnal1, 0)
test2$HPC.P14 <- ifelse(test2$TimePoint == "HPC.P14", test2$Ctnnal1, 0)
test2$HPC.Adult <- ifelse(test2$TimePoint == "HPC.Adult", test2$Ctnnal1, 0)


aCatadult <- ggplot(test2, aes(x = Cell, y = HPC.Adult, fill = as.factor(Cluster))) + 
    geom_bar(stat = "identity", width = 1) +
  theme(
    axis.text = element_blank(), 
    axis.title.x = element_blank(), 
    axis.ticks = element_blank(), 
    legend.position = "none",
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1)
  ) + 
  labs(y="Adult")
 

aCate16 <- ggplot(test2, aes(x = Cell, y = HPC.E16, fill = as.factor(Cluster))) + 
    geom_bar(stat = "identity", width = 1) +
  theme(
    axis.text = element_blank(), 
    axis.title.x = element_blank(), 
    axis.ticks = element_blank(),
    legend.position = "none", 
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1)
  ) + 
  labs(y="E16.5")

aCatp14 <- ggplot(test2, aes(x = Cell, y = HPC.P14, fill = as.factor(Cluster))) + 
    geom_bar(stat = "identity", width = 1) +
  theme(
    axis.text = element_blank(), 
    axis.title.x = element_blank(), 
    axis.ticks = element_blank(), 
    legend.position = "none", 
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1)
  ) + 
  labs(y="P14")

aCatp7 <- ggplot(test2, aes(x = Cell, y = HPC.P7, fill = as.factor(Cluster))) + 
    geom_bar(stat = "identity", width = 1) +
  theme(
    axis.text = element_blank(), 
    axis.title.x = element_blank(), 
    axis.ticks = element_blank(), 
    legend.position = "none",
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1)
  ) + 
  labs(y="P7")



barchart <- aCate16 / aCatp7 / aCatp14 / aCatadult

barchart

```

barchart <- aCatbar / cd9bar
barchart

    axis.text = element_blank(), 
    axis.title = element_blank(), 
    axis.ticks = element_blank(), 
    legend.position = "none",


```{r}
vlnage <- VlnPlot(so, features = c("Ctnnal1"), group.by = "orig.ident")
ggsave("ctnnal1vlnplotage.png", vlnage, height = 10, width = 8, dpi = 600)

RidgePlot(so, features = c("Ctnnal1"), group.by = "orig.ident")

```

```{r}
Idents(so) <- "seurat_clusters"
c10 <- FindMarkers(so, ident.1 = 10, ident.2 = NULL)

```

```{r}
x <- FetchData(so, vars = c("Ctnnal1"))
so$acat <- ifelse(x$Ctnnal1 > 0, TRUE, FALSE)
table(so$acat)

aCat <- subset(so, subset = Ctnnal1 > 0)
Idents(aCat) <- "orig.ident"
degtest <- FindMarkers(aCat, ident.1 = "HPC.E16.5", ident.2 = "HPC.Adult")

```

```{r GFP+ signature}
gfpsig <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/RNASeq/Gene Signatures/gfpposoverallsignature.csv")
gfpsig <- gfpsig[,-1]

gfppn <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/RNASeq/Gene Signatures/gfppospnsignature.csv")
gfppn <- gfppn[,-1]

Idents(so) <- "seurat_clusters"
so$orig.ident <- factor(so$orig.ident, levels = c("HPC.E16.5", "HPC.P7", "HPC.P14", "HPC.Adult"))
so <- AddModuleScore(so, features = list(gfpsig), name = "GFPpos")
FeaturePlot(so, features = "GFPpos1")
FeaturePlot(so, features = "GFPpos1", split.by = "orig.ident")
VlnPlot(so, features = "GFPpos1")
VlnPlot(so, features = "GFPpos1", split.by = "orig.ident")
RidgePlot(so, features = "GFPpos1", group.by = "orig.ident")


so <- AddModuleScore(so, features = list(gfppn), name = "GFPposPN")
pnmodfeat <-  FeaturePlot(so, features = "GFPposPN1")
pnmodfeat
ggsave("pngfppossigfeatureplot.png", pnmodfeat, height = 8, width = 10, dpi = 600)
pnmodfeatage <-  FeaturePlot(so, features = "GFPposPN1", split.by = "orig.ident")
pnmodfeatage
ggsave("pngfppossigfeatureplotage.png", pnmodfeatage, height = 8, width = 30, dpi = 600)
VlnPlot(so, features = "GFPposPN1")
VlnPlot(so, features = "GFPposPN1", group.by = "orig.ident")
RidgePlot(so, features = "GFPposPN1", group.by = "orig.ident")
Idents(so) <- "acat"
gfpvln <- VlnPlot(so, features = "GFPposPN1", pt.size = 0) + 
  geom_jitter(width=0.1, size=0.5)
gfpvlnsplit <- VlnPlot(so, features = "GFPposPN1", split.by = "orig.ident")
ggsave("gfpposvlnage.png", gfpvlnsplit, height = 10, width = 8, dpi = 600)
ggsave("gfpposvln.png", gfpvln, height = 10, width = 8, dpi = 600)

ggplot(so, aes(x=so$GFPpos1)) + 
  geom_histogram()

gfpmodule <- so$GFPpos1
gfpmoduledf <- data.frame(gfpmodule)
ggplot(gfpmoduledf, aes(x=gfpmodule)) + 
  geom_histogram()

```

```{r acat stats exploration}
E16 <- subset(so, orig.ident == "HPC.E16.5")
P7 <- subset(so, orig.ident == "HPC.P7")
P14 <- subset(so, orig.ident == "HPC.P14")
Adult <- subset(so, orig.ident == "HPC.Adult")

gfpe16 <- median(E16$GFPpos1)
gfpp7 <- median(P7$GFPpos1)
gfpp14 <- median(P14$GFPpos1)
gfpadult <- median(Adult$GFPpos1)

mediansignature <- data.frame(E16 = gfpe16,
                              P7 = gfpp7, 
                              P14 = gfpp14, 
                              Adult = gfpadult)

gfpe16 <- median(E16$GFPposPN1)
gfpp7 <- median(P7$GFPposPN1)
gfpp14 <- median(P14$GFPposPN1)
gfpadult <- median(Adult$GFPposPN1)

mediansignaturepn <- data.frame(E16 = gfpe16,
                              P7 = gfpp7, 
                              P14 = gfpp14, 
                              Adult = gfpadult)
acattable <- data.frame(E16 = sum(E16$acat)/length(E16$acat),
                        P7 = sum(P7$acat)/length(P7$acat),
                        P14 = sum(P14$acat)/length(P14$acat),
                        Adult = sum(Adult$acat)/length(Adult$acat))

acatlong <- data.frame(
  Population = c("E16", "P7", "P14", "Adult"),
  Proportion = c(sum(E16$acat)/length(E16$acat),
                 sum(P7$acat)/length(P7$acat),
                 sum(P14$acat)/length(P14$acat),
                 sum(Adult$acat)/length(Adult$acat)))

write.csv(acattable, "percentctnnal1pos.csv")

E16acat <- subset(E16, acat == TRUE)
P7acat <- subset(P7, acat == TRUE)
P14acat <- subset(P14, acat == TRUE)
Adultacat <- subset(Adult, acat == TRUE)

ctnnal <- subset(so, acat == TRUE)
x1 <- FetchData(E16acat, vars = c("Ctnnal1"))
x2 <- FetchData(P7acat, vars = c("Ctnnal1"))
x3 <- FetchData(P14acat, vars = c("Ctnnal1"))
x4 <- FetchData(Adultacat, vars = c("Ctnnal1"))
x1$Population <- "E16"
x2$Population <- "P7"
x3$Population <- "P14"
x4$Population <- "Adult"
xdf <- data.frame(E16 = x1$Ctnnal1,
                  P7 = x2$Ctnnal1, 
                  P14 = x3$Ctnnal1,
                  Adult = x4$Ctnnal1)
xdf <- rbind(x1, x2, x3, x4)

xdf$Population <- factor(xdf$Population, levels = c("E16", "P7", "P14", "Adult"))
boxplot <- ggplot(xdf, aes(x = Population, y = Ctnnal1, fill = Population)) + 
  geom_boxplot()
boxplot
ggsave("ctnnal1boxplot.png", boxplot, height = 10, width = 8, dpi = 600)

acatlong$Population <- factor(acatlong$Population, levels = c("E16", "P7", "P14", "Adult"))
ggplot(acatlong, aes(x=Population, y=Proportion, fill = Population)) + 
  geom_col() + 
  labs(title = "Ctnnal1 Positive Cells") + 
  theme_minimal() + 
  theme(plot.title = element_text(hjust = 0.5),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())

rm(aCatadult, aCatp7, aCatp14, aCate16, aCatbar, acattable, acatlong, Adult, Adultacat, barchart, E16, E16acat, P7, P7acat, P14, P14acat, test2, test2p, testplot, vlnage)
rm(ctnnal, gfpmoduledf, x, x1, x2, x3, x4, xdf, ydf)
rm(boxplot, gfpvln, gfpvlnsplit, heatmap, mediansignature, mediansignaturepn)

```


```{r Heatmap}
negative <- read.csv("C:/Users/rohit/OneDrive - Loyola University Chicago/Zhang Lab/RNASeq/Gene Signatures/youngnegativedeg.csv")
negative <- negative[,-1]
genelist <- c(gfppn, negative)

genes <- read.csv("singlecellheatmapgenelist3.csv")

so <- SCTransform(so, assay = "RNA", new.assay.name = "SCT", variable.features.n = nrow(so), verbose = TRUE)
so$acatage <- paste(so$acat, so$orig.ident, sep = "_")
so$acatage <- factor(so$acatage, levels = c("FALSE_HPC.E16.5", "TRUE_HPC.E16.5", "FALSE_HPC.P7", "TRUE_HPC.P7", "FALSE_HPC.P14", "TRUE_HPC.P14", "FALSE_HPC.Adult", "TRUE_HPC.Adult"))
so$acatcluster <- paste(so$acat, so$seurat_clusters, sep = "_")

# Create a heatmap
heatmap <- DoHeatmap(
  object = so,
  assay = "SCT",
  features = genes$x,
  group.by = "acatage",
  lines.width = 20
) + scale_fill_gradientn(colors = c("blue", "white", "red"), values = c(0, 0.5, 1),
  rescaler = function(x, to = c(0, 1), from = range(x, na.rm = TRUE, finite = TRUE)) {
    scales::rescale_mid(x, mid = 0, to = to, from = from)
  })
heatmap

ggsave("hpcheatmap4v5.png", heatmap, height = 15, width = 15, dpi = 600)

```

```{r QC}
missing <- c("Scin", "Adcyap1r1", "Acmsd", "Myo18b", "Gm44154", "X1110032F04Rik", "Gm44954", "Zfp175", "Gm38825", "Fam241b", "Sertm2", "X5830416I19Rik", "Gucy2e", "C430002N11Rik", "Gpr15", "Pax9", "Mexis", "Igkv6.32", "Ighv1.80", "Trem6l", "Tpsb2", "Gm3360", "Adra2a", "Ccbe1", "Chsy3", "Mtss2", "Tmem217rt", "Lhfpl6")

missing2 <- c("Gm44154", "X1110032F04Rik", "Gm44954", "Zfp175", "Gm38825", "Fam241b", "Sertm2", "X5830416I19Rik", "C430002N11Rik", "Mexis", "Igkv6.32", "Ighv1.80", "Trem6l", "Gm3360", "Mtss2", "Tmem217rt", "Lhfpl6")

missing3 <- c("Rnf39", "Adcyap1r1", "Acmsd", "Gm44154", "Pcsk6", "X1110032F04Rik", "Gm44954", "Zfp175", "Gm38825", "Fam241b", "Sertm2", "X5830416I19Rik", "Gucy2e", "C430002N11Rik", "Gpr15", "Pax9", "Mexis", "Igkv6.32", "Ighv1.80", "Trem6l", "Gm3360", "Adra2a", "Ccbe1", "Chsy3", "Gpx8", "Amotl2", "Pls1", "Mtss2", "Tmem217rt", "Lhfpl6")
#30 total, 9 positive and 21 negative

posmissing <- Reduce(intersect, list(missing3, gfppn))
negmissing <- Reduce(intersect, list(missing3, negative))

```

```{r trying things 06-29-2025}
# Create a binary label for Ctnnal1 expression
so$Ctnnal1_status <- ifelse(GetAssayData(so, assay = "RNA", slot = "data")["Ctnnal1", ] > 0, "Ctnnal1+", "Ctnnal1-")

# Subset Ctnnal1+ cells from HSC.E16.5
group1_cells <- WhichCells(so, expression = orig.ident == "HSC.E16.5" & Ctnnal1_status == "Ctnnal1+")

# Subset all cells from HSC.Adult
group2_cells <- WhichCells(so, expression = orig.ident == "HSC.Adult")

# Run differential expression
markers <- FindMarkers(so, ident.1 = group1_cells, ident.2 = group2_cells, logfc.threshold = 0.25)

# View top markers
head(markers)

# Subset Ctnnal1+ cells from HSC.E16.5
group1_cells <- WhichCells(so, expression = orig.ident == "HSC.E16.5" & Ctnnal1_status == "Ctnnal1+")

# Subset Ctnnal1- cells from HSC.E16.5
group2_cells <- WhichCells(so, expression = orig.ident == "HSC.E16.5" & Ctnnal1_status == "Ctnnal1-")

# Run differential expression
markers_ctnnal1_e16.5 <- FindMarkers(so, ident.1 = group1_cells, ident.2 = group2_cells, logfc.threshold = 0.25)

# View top markers
head(markers_ctnnal1_e16.5)

# Subset Ctnnal1− cells from HSC.E16.5
group_neg_e16 <- WhichCells(so, expression = orig.ident == "HSC.E16.5" & Ctnnal1_status == "Ctnnal1-")

# Subset all cells from HSC.Adult
group_adult <- WhichCells(so, expression = orig.ident == "HSC.Adult")

# Run differential expression
markers_neg_vs_adult <- FindMarkers(so, ident.1 = group_neg_e16, ident.2 = group_adult, logfc.threshold = 0.25)

# View top markers
head(markers_adult_vs_neg)

# Create a table of Ctnnal1 status by orig.ident
ctnnal1_table <- table(so$orig.ident, so$Ctnnal1_status)

# Convert to a data frame for easier viewing or export
ctnnal1_df <- as.data.frame(ctnnal1_table)

# View the table
print(ctnnal1_df)

write.csv(markers_ctnnal1_e16.5, "markers_ctnnal1_e16.csv")
write.csv(markers, "markers_ctnnal1e16vsadult.csv")

run_gsea(markers_ctnnal1_e16.5, "m1.all.v2025.1.Mm.symbols.gmt", "e16acatchr_gsea.csv")
run_gsea(markers_ctnnal1_e16.5, "m3.gtrd.v2025.1.Mm.symbols.gmt", "e16acattf_gsea.csv")

run_gsea(markers_neg_vs_adult, "m1.all.v2025.1.Mm.symbols.gmt", "e16negvadultchr_gsea.csv")
run_gsea(markers_neg_vs_adult, "m3.gtrd.v2025.1.Mm.symbols.gmt", "e16negvadulttf_gsea.csv")

run_gsea(markers, "m1.all.v2025.1.Mm.symbols.gmt", "e16posvadultchr_gsea.csv")
run_gsea(markers, "m3.gtrd.v2025.1.Mm.symbols.gmt", "e16posvadulttf_gsea.csv")

```

```{r p14vadult}
# Create a binary label for Ctnnal1 expression
so$Ctnnal1_status <- ifelse(GetAssayData(so, assay = "RNA", slot = "data")["Ctnnal1", ] > 0, "Ctnnal1+", "Ctnnal1-")

# Subset Ctnnal1+ cells from HSC.P14
group1_cells <- WhichCells(so, expression = orig.ident == "HSC.P14" & Ctnnal1_status == "Ctnnal1+")

# Subset all cells from HSC.Adult
group2_cells <- WhichCells(so, expression = orig.ident == "HSC.Adult")

# Run differential expression
markers <- FindMarkers(so, ident.1 = group1_cells, ident.2 = group2_cells, logfc.threshold = 0.25)

# View top markers
head(markers)


# Subset Ctnnal1+ cells from HSC.P14
group1_cells <- WhichCells(so, expression = orig.ident == "HSC.P14" & Ctnnal1_status == "Ctnnal1+")

# Subset Ctnnal1− cells from HSC.P14
group2_cells <- WhichCells(so, expression = orig.ident == "HSC.P14" & Ctnnal1_status == "Ctnnal1-")

# Run differential expression
markers_ctnnal1_p14 <- FindMarkers(so, ident.1 = group1_cells, ident.2 = group2_cells, logfc.threshold = 0.25)

# View top markers
head(markers_ctnnal1_p14)

# Subset Ctnnal1− cells from HSC.P14
group_neg_p14 <- WhichCells(so, expression = orig.ident == "HSC.P14" & Ctnnal1_status == "Ctnnal1-")

# Subset all cells from HSC.Adult
group_adult <- WhichCells(so, expression = orig.ident == "HSC.Adult")

# Run differential expression
markers_neg_vs_adult <- FindMarkers(so, ident.1 = group_neg_p14, ident.2 = group_adult, logfc.threshold = 0.25)

# View top markers
head(markers_adult_vs_neg)

write.csv(markers_ctnnal1_p14, "markers_ctnnal1_p14.csv")
write.csv(markers, "markers_ctnnal1p14vsadult.csv")

run_gsea(markers_ctnnal1_p14, "m1.all.v2025.1.Mm.symbols.gmt", "p14acatchr_gsea.csv")
run_gsea(markers_ctnnal1_p14, "m3.gtrd.v2025.1.Mm.symbols.gmt", "p14acattf_gsea.csv")

run_gsea(markers_neg_vs_adult, "m1.all.v2025.1.Mm.symbols.gmt", "p14negvadultchr_gsea.csv")
run_gsea(markers_neg_vs_adult, "m3.gtrd.v2025.1.Mm.symbols.gmt", "p14negvadulttf_gsea.csv")

run_gsea(markers, "m1.all.v2025.1.Mm.symbols.gmt", "p14posvadultchr_gsea.csv")
run_gsea(markers, "m3.gtrd.v2025.1.Mm.symbols.gmt", "p14posvadulttf_gsea.csv")

```

```{r}
gene_of_interest <- "Ctnnal1"  # for example
expr_matrix <- FetchData(so, vars = c(gene_of_interest, rownames(so)))
cor_scores <- apply(expr_matrix[ , -1], 2, function(x) cor(x, expr_matrix[ , 1], method = "pearson"))
top_correlated <- sort(cor_scores, decreasing = TRUE)[1:50]
low_cor <- sort(cor_scores, decreasing = FALSE)[1:50]

top_genes <- sort(cor_scores, decreasing = TRUE)[1:30]
low_genes <- sort(cor_scores, decreasing = FALSE)[1:30]
gene_names <- names(top_genes)
expr_data <- FetchData(so, vars = gene_names)
pheatmap(t(scale(expr_data)), cluster_rows = TRUE, cluster_cols = TRUE)

barplot(top_genes, las = 2, col = "steelblue", main = "Top Correlated Genes")
barplot(low_genes, las = 2, col = "steelblue", main = "Lowest Correlated Genes")

FeatureScatter(so, feature1 = "Ctnnal1", feature2 = "Mllt3")

```

# HSC Dataset

```{r libraries, results='hide'}
library(Seurat)
library(ggplot2)
library(patchwork)
library(dplyr)
library(biomaRt)
library(pheatmap)
library(fgsea)
library(patchwork)
library(scales)
library(ggbreak)
library(ggcyto)
library(DESeq2)
options(future.globals.maxSize = 10000 * 1024^2)
setwd("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/RNASeq/HSC GSE128761")

run_gsea <- function(markers, gmt_path, output_gsea) {
rnk <- setNames(markers$avg_log2FC, rownames(markers))
rnk <- rnk[!is.na(rnk)]
gmt <- gmtPathways(gmt_path)
gsea <- fgsea(pathways = gmt, stats = rnk, minSize = 15, maxSize = 500)
gseadf <- as.data.frame(gsea)
list_columns <- sapply(gseadf, is.list)
gseadf[list_columns] <- lapply(gseadf[list_columns], function(col) sapply(col, paste, collapse = ";"))
write.csv(gseadf, output_gsea)
}

EnsIDReplace2 <- function(input, output){
  gene_mapping <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/RNASeq/Cite-seq/GSE145491/Ensembl_labels.csv")
  rna <- input
  rnanames <- rownames(rna)
  id_list <- setNames(gene_mapping$gene_name, gene_mapping$gene_id)
  rnaname <- character(nrow(rna))
  for (i in seq_len(nrow(rna))) {
    ensembl_id <- rownames(rna)[i]
    if (ensembl_id %in% names(id_list)) {
      rnaname[i] <- id_list[[ensembl_id]]
    } else{
      rnaname[i] <- ensembl_id
    }
  }
  rnaname[rnaname == ""] <- NA
  
  rownames(rna) <- rnaname
  output <- rna
  rm(gene_mapping, id_list, ensembl_id, i, rnaname)
}
```


``` {r Load Seurat Object and Cluster}
data <- Read10X(data.dir = "HSC_P7/")
so <- CreateSeuratObject(counts = data)
so <- NormalizeData(so)
so <- FindVariableFeatures(so)
so <- ScaleData(so)
so <- RunPCA(so)
so <- FindNeighbors(so, dims = 1:30)
so <- FindClusters(so)
so <- RunUMAP(so, dims = 1:30)
rm(data)
```

```{r}
so <- HSC_joined
rm(HSC_joined, HSC.markers)
so$orig.ident <- factor(so$orig.ident, levels = c("HSC.E16.5", "HSC.P7", "HSC.P14", "HSC.Adult"))

DimPlot(so, group.by = "orig.ident")
FeaturePlot(so, features = c("Ctnnal1"))
dimacat <- DimPlot(so, group.by = "acat", cols = c("grey", "blue"))
dimacat
ggsave("hscumapacat.png", dimacat, height = 8, width = 10, dpi = 600)
DimPlot(so, group.by = "orig.ident")
dimplot <- DimPlot(so)
ggsave("hscumap.png", dimplot, height = 8, width = 10, dpi = 600)

dimtime <- DimPlot(so, group.by = "orig.ident")
dimtime
ggsave("hscumapage.png", dimtime, height = 8, width = 10, dpi = 600)

FeaturePlot(so, features = c("Ctnnal1"))
```

```{r trying things 06-27-2025}
# Create a binary label for Ctnnal1 expression
so$Ctnnal1_status <- ifelse(GetAssayData(so, assay = "RNA", slot = "data")["Ctnnal1", ] > 0, "Ctnnal1+", "Ctnnal1-")

# Subset Ctnnal1+ cells from HSC.E16.5
group1_cells <- WhichCells(so, expression = orig.ident == "HSC.E16.5" & Ctnnal1_status == "Ctnnal1+")

# Subset all cells from HSC.Adult
group2_cells <- WhichCells(so, expression = orig.ident == "HSC.Adult")

# Run differential expression
markers <- FindMarkers(so, ident.1 = group1_cells, ident.2 = group2_cells, logfc.threshold = 0.25)

# View top markers
head(markers)

# Subset Ctnnal1+ cells from HSC.E16.5
group1_cells <- WhichCells(so, expression = orig.ident == "HSC.E16.5" & Ctnnal1_status == "Ctnnal1+")

# Subset Ctnnal1- cells from HSC.E16.5
group2_cells <- WhichCells(so, expression = orig.ident == "HSC.E16.5" & Ctnnal1_status == "Ctnnal1-")

# Run differential expression
markers_ctnnal1_e16.5 <- FindMarkers(so, ident.1 = group1_cells, ident.2 = group2_cells, logfc.threshold = 0.25)

# View top markers
head(markers_ctnnal1_e16.5)

# Subset Ctnnal1− cells from HSC.E16.5
group_neg_e16 <- WhichCells(so, expression = orig.ident == "HSC.E16.5" & Ctnnal1_status == "Ctnnal1-")

# Subset all cells from HSC.Adult
group_adult <- WhichCells(so, expression = orig.ident == "HSC.Adult")

# Run differential expression
markers_adult_vs_neg <- FindMarkers(so, ident.1 = group_adult, ident.2 = group_neg_e16, logfc.threshold = 0.25)

# View top markers
head(markers_adult_vs_neg)

# Create a table of Ctnnal1 status by orig.ident
ctnnal1_table <- table(so$orig.ident, so$Ctnnal1_status)

# Convert to a data frame for easier viewing or export
ctnnal1_df <- as.data.frame(ctnnal1_table)

# View the table
print(ctnnal1_df)

write.csv(markers_ctnnal1_e16.5, "markers_ctnnal1_e16.csv")
write.csv(markers, "markers_ctnnal1e16vsadult.csv")

run_gsea(markers_ctnnal1_e16.5, "m3.mirdb.v2025.1.Mm.symbols.gmt", "e16acatmirna_gsea.csv")

run_gsea(markers_adult_vs_neg, "m3.mirdb.v2025.1.Mm.symbols.gmt", "e16negvadultmirna_gsea.csv")

run_gsea(markers, "m3.mirdb.v2025.1.Mm.symbols.gmt", "e16posvadultmirna_gsea.csv")

run_gsea(markers_ctnnal1_e16.5, "m1.all.v2025.1.Mm.symbols.gmt", "e16acatchr_gsea.csv")
run_gsea(markers_ctnnal1_e16.5, "m3.gtrd.v2025.1.Mm.symbols.gmt", "e16acattf_gsea.csv")

run_gsea(markers_adult_vs_neg, "m1.all.v2025.1.Mm.symbols.gmt", "e16negvadultchr_gsea.csv")
run_gsea(markers_adult_vs_neg, "m3.gtrd.v2025.1.Mm.symbols.gmt", "e16negvadulttf_gsea.csv")

run_gsea(markers, "m1.all.v2025.1.Mm.symbols.gmt", "e16posvadultchr_gsea.csv")
run_gsea(markers, "m3.gtrd.v2025.1.Mm.symbols.gmt", "e16posvadulttf_gsea.csv")

```

```{r trying things Hmga2 07-03-2025}
# Create a binary label for Hmga2 expression
so$Hmga2_status <- ifelse(GetAssayData(so, assay = "RNA", slot = "data")["Hmga2", ] > 0, "Hmga2+", "Hmga2-")

# Subset Hmga2+ cells from HSC.E16.5
group1_cells <- WhichCells(so, expression = orig.ident == "HSC.E16.5" & Hmga2_status == "Hmga2+")

# Subset all cells from HSC.Adult
group2_cells <- WhichCells(so, expression = orig.ident == "HSC.Adult")

# Run differential expression
markers <- FindMarkers(so, ident.1 = group1_cells, ident.2 = group2_cells, logfc.threshold = 0.25)

# View top markers
head(markers)

# Subset Hmga2+ cells from HSC.E16.5
group1_cells <- WhichCells(so, expression = orig.ident == "HSC.E16.5" & Hmga2_status == "Hmga2+")

# Subset Hmga2- cells from HSC.E16.5
group2_cells <- WhichCells(so, expression = orig.ident == "HSC.E16.5" & Hmga2_status == "Hmga2-")

# Run differential expression
markers_Hmga2_e16.5 <- FindMarkers(so, ident.1 = group1_cells, ident.2 = group2_cells, logfc.threshold = 0.25)

# View top markers
head(markers_Hmga2_e16.5)

# Subset Hmga2− cells from HSC.E16.5
group_neg_e16 <- WhichCells(so, expression = orig.ident == "HSC.E16.5" & Hmga2_status == "Hmga2-")

# Subset all cells from HSC.Adult
group_adult <- WhichCells(so, expression = orig.ident == "HSC.Adult")

# Run differential expression
markers_adult_vs_neg <- FindMarkers(so, ident.1 = group_adult, ident.2 = group_neg_e16, logfc.threshold = 0.25)

# View top markers
head(markers_adult_vs_neg)

# Create a table of Hmga2 status by orig.ident
Hmga2_table <- table(so$orig.ident, so$Hmga2_status)

# Convert to a data frame for easier viewing or export
Hmga2_df <- as.data.frame(Hmga2_table)

# View the table
print(Hmga2_df)

write.csv(markers_Hmga2_e16.5, "markers_Hmga2_e16.csv")
write.csv(markers, "markers_Hmga2e16vsadult.csv")

run_gsea(markers_Hmga2_e16.5, "m3.mirdb.v2025.1.Mm.symbols.gmt", "e16hmga2mirna_gsea.csv")

run_gsea(markers_adult_vs_neg, "m3.mirdb.v2025.1.Mm.symbols.gmt", "e16hmga2negvadultmirna_gsea.csv")

run_gsea(markers, "m3.mirdb.v2025.1.Mm.symbols.gmt", "e16posvadultmirna_hmga2_gsea.csv")

run_gsea(markers_Hmga2_e16.5, "m1.all.v2025.1.Mm.symbols.gmt", "e16hmga2chr_gsea.csv")
run_gsea(markers_Hmga2_e16.5, "m3.gtrd.v2025.1.Mm.symbols.gmt", "e16hmga2tf_gsea.csv")

run_gsea(markers_adult_vs_neg, "m1.all.v2025.1.Mm.symbols.gmt", "e16negvadultchr_hmga2_gsea.csv")
run_gsea(markers_adult_vs_neg, "m3.gtrd.v2025.1.Mm.symbols.gmt", "e16negvadulttf_hmga2_gsea.csv")

run_gsea(markers, "m1.all.v2025.1.Mm.symbols.gmt", "e16posvadultchr_hmga2_gsea.csv")
run_gsea(markers, "m3.gtrd.v2025.1.Mm.symbols.gmt", "e16posvadulttf_hmga2_gsea.csv")

```

```{r p14vadult}
# Create a binary label for Ctnnal1 expression
so$Ctnnal1_status <- ifelse(GetAssayData(so, assay = "RNA", slot = "data")["Ctnnal1", ] > 0, "Ctnnal1+", "Ctnnal1-")

# Subset Ctnnal1+ cells from HSC.P14
group1_cells <- WhichCells(so, expression = orig.ident == "HSC.P14" & Ctnnal1_status == "Ctnnal1+")

# Subset all cells from HSC.Adult
group2_cells <- WhichCells(so, expression = orig.ident == "HSC.Adult")

# Run differential expression
markers <- FindMarkers(so, ident.1 = group1_cells, ident.2 = group2_cells, logfc.threshold = 0.25)

# View top markers
head(markers)

# Subset Ctnnal1+ cells from HSC.P14
group1_cells <- WhichCells(so, expression = orig.ident == "HSC.P14" & Ctnnal1_status == "Ctnnal1+")

# Subset Ctnnal1− cells from HSC.P14
group2_cells <- WhichCells(so, expression = orig.ident == "HSC.P14" & Ctnnal1_status == "Ctnnal1-")

# Run differential expression
markers_ctnnal1_p14 <- FindMarkers(so, ident.1 = group1_cells, ident.2 = group2_cells, logfc.threshold = 0.25)

# View top markers
head(markers_ctnnal1_p14)

# Subset Ctnnal1− cells from HSC.P14
group_neg_p14 <- WhichCells(so, expression = orig.ident == "HSC.P14" & Ctnnal1_status == "Ctnnal1-")

# Subset all cells from HSC.Adult
group_adult <- WhichCells(so, expression = orig.ident == "HSC.Adult")

# Run differential expression
markers_adult_vs_neg <- FindMarkers(so, ident.1 = group_adult, ident.2 = group_neg_p14, logfc.threshold = 0.25)

# View top markers
head(markers_adult_vs_neg)

write.csv(markers_ctnnal1_p14, "markers_ctnnal1_p14.csv")
write.csv(markers, "markers_ctnnal1p14vsadult.csv")

run_gsea(markers_ctnnal1_p14, "m3.mirdb.v2025.1.Mm.symbols.gmt", "p14acatmirna_gsea.csv")

run_gsea(markers_adult_vs_neg, "m3.mirdb.v2025.1.Mm.symbols.gmt", "p14negvadultmirna_gsea.csv")

run_gsea(markers, "m3.mirdb.v2025.1.Mm.symbols.gmt", "p14posvadultmirna_gsea.csv")



run_gsea(markers_ctnnal1_p14, "m1.all.v2025.1.Mm.symbols.gmt", "p14acatchr_gsea.csv")
run_gsea(markers_ctnnal1_p14, "m3.gtrd.v2025.1.Mm.symbols.gmt", "p14acattf_gsea.csv")

run_gsea(markers_adult_vs_neg, "m1.all.v2025.1.Mm.symbols.gmt", "p14negvadultchr_gsea.csv")
run_gsea(markers_adult_vs_neg, "m3.gtrd.v2025.1.Mm.symbols.gmt", "p14negvadulttf_gsea.csv")

run_gsea(markers, "m1.all.v2025.1.Mm.symbols.gmt", "p14posvadultchr_gsea.csv")
run_gsea(markers, "m3.gtrd.v2025.1.Mm.symbols.gmt", "p14posvadulttf_gsea.csv")

```

```{r p14vadult Hmga2}
# Create a binary label for Hmga2 expression
so$Hmga2_status <- ifelse(GetAssayData(so, assay = "RNA", slot = "data")["Hmga2", ] > 0, "Hmga2+", "Hmga2-")

# Subset Hmga2+ cells from HSC.P14
group1_cells <- WhichCells(so, expression = orig.ident == "HSC.P14" & Hmga2_status == "Hmga2+")

# Subset all cells from HSC.Adult
group2_cells <- WhichCells(so, expression = orig.ident == "HSC.Adult")

# Run differential expression
markers <- FindMarkers(so, ident.1 = group1_cells, ident.2 = group2_cells, logfc.threshold = 0.25)

# View top markers
head(markers)

# Subset Hmga2+ cells from HSC.P14
group1_cells <- WhichCells(so, expression = orig.ident == "HSC.P14" & Hmga2_status == "Hmga2+")

# Subset Hmga2− cells from HSC.P14
group2_cells <- WhichCells(so, expression = orig.ident == "HSC.P14" & Hmga2_status == "Hmga2-")

# Run differential expression
markers_Hmga2_p14 <- FindMarkers(so, ident.1 = group1_cells, ident.2 = group2_cells, logfc.threshold = 0.25)

# View top markers
head(markers_Hmga2_p14)

# Subset Hmga2− cells from HSC.P14
group_neg_p14 <- WhichCells(so, expression = orig.ident == "HSC.P14" & Hmga2_status == "Hmga2-")

# Subset all cells from HSC.Adult
group_adult <- WhichCells(so, expression = orig.ident == "HSC.Adult")

# Run differential expression
markers_adult_vs_neg <- FindMarkers(so, ident.1 = group_adult, ident.2 = group_neg_p14, logfc.threshold = 0.25)

# View top markers
head(markers_adult_vs_neg)

write.csv(markers_Hmga2_p14, "markers_Hmga2_p14.csv")
write.csv(markers, "markers_Hmga2p14vsadult.csv")

run_gsea(markers_Hmga2_p14, "m3.mirdb.v2025.1.Mm.symbols.gmt", "p14Hmga2mirna_gsea.csv")

run_gsea(markers_adult_vs_neg, "m3.mirdb.v2025.1.Mm.symbols.gmt", "p14negvadultmirna_hmga2_gsea.csv")

run_gsea(markers, "m3.mirdb.v2025.1.Mm.symbols.gmt", "p14posvadultmirna_hmga2_gsea.csv")



run_gsea(markers_Hmga2_p14, "m1.all.v2025.1.Mm.symbols.gmt", "p14Hmga2chr_gsea.csv")
run_gsea(markers_Hmga2_p14, "m3.gtrd.v2025.1.Mm.symbols.gmt", "p14Hmga2tf_gsea.csv")

run_gsea(markers_adult_vs_neg, "m1.all.v2025.1.Mm.symbols.gmt", "p14negvadultchr_hmga2_gsea.csv")
run_gsea(markers_adult_vs_neg, "m3.gtrd.v2025.1.Mm.symbols.gmt", "p14negvadulttf_hmga2_gsea.csv")

run_gsea(markers, "m1.all.v2025.1.Mm.symbols.gmt", "p14posvadultchr_hmga2_gsea.csv")
run_gsea(markers, "m3.gtrd.v2025.1.Mm.symbols.gmt", "p14posvadulttf_hmga2_gsea.csv")

```

```{r}
gene_of_interest <- "Ctnnal1"  # for example
expr_matrix <- FetchData(so, vars = c(gene_of_interest, rownames(so)))
cor_scores <- apply(expr_matrix[ , -1], 2, function(x) cor(x, expr_matrix[ , 1], method = "pearson"))
top_correlated <- sort(cor_scores, decreasing = TRUE)[1:50]
low_cor <- sort(cor_scores, decreasing = FALSE)[1:50]

top_genes <- sort(cor_scores, decreasing = TRUE)[1:30]
low_genes <- sort(cor_scores, decreasing = FALSE)[1:30]
gene_names <- names(top_genes)
expr_data <- FetchData(so, vars = gene_names)
pheatmap(t(scale(expr_data)), cluster_rows = TRUE, cluster_cols = TRUE)

barplot(top_genes, las = 2, col = "steelblue", main = "Top Correlated Genes")
barplot(low_genes, las = 2, col = "steelblue", main = "Lowest Correlated Genes")


FeatureScatter(so, feature1 = "Ctnnal1", feature2 = "Hmga2")

```

```{r}
# Ensure factor levels are ordered
so$orig.ident <- factor(so$orig.ident, levels = c("HSC.E16.5", "HSC.P7", "HSC.P14", "HSC.Adult"))

# Subset to just the identities of interest
hsc_subset <- subset(so, subset = orig.ident %in% levels(so$orig.ident))

# Define Ctnnal1 expression status (expression > 0)
hsc_subset$Ctnnal1_status <- ifelse(FetchData(hsc_subset, "Ctnnal1")[,1] > 0, "Ctnnal1+", "Ctnnal1-")

# Combine orig.ident and status for stratification
hsc_subset$Identity_Status <- paste(hsc_subset$orig.ident, hsc_subset$Ctnnal1_status, sep = "_")
Idents(hsc_subset) <- hsc_subset$Identity_Status

# Run PCA (make sure you've scaled data beforehand)
hsc_subset <- RunPCA(hsc_subset, features = VariableFeatures(hsc_subset))

# Plot PCA with split identities
DimPlot(hsc_subset, reduction = "pca", group.by = "Identity_Status", pt.size = 1) +
  ggtitle("PCA: Ctnnal1+ vs - within HSC Developmental Stages") +
  theme(axis.title = element_text(size = 12), plot.title = element_text(hjust = 0.5))


```

```{r}
goi <- c("Ctnnal1", "Hmga2", "Hhex", "Mllt3")
soe16 <- subset(so, so$orig.ident == "HSC.E16.5")
sop7 <- subset(so, so$orig.ident == "HSC.P7")
sop14 <- subset(so, so$orig.ident == "HSC.P14")
soadult <- subset(so, so$orig.ident == "HSC.Adult")
data <- FetchData(soadult, vars = goi)
datafil <- data[data$Ctnnal1 > 0, ]
datafil <- datafil[datafil$Hmga2 > 0, ]
datafil <- datafil[datafil$Hhex > 0, ]
datafil <- datafil[datafil$Mllt3 > 0, ]

cor_vec <- sapply(colnames(data)[-1], function(gene) {
  cor(data$Ctnnal1, data[[gene]], method = "pearson")
})

ggplot(data, aes(x = Ctnnal1, y = Mllt3)) +
  geom_point() +
  labs(
    title = "Ctnnal1 vs Mllt3 Adult",
    x = "Ctnnal1 (raw counts)",
    y = "Mllt3 (raw counts)"
  ) +
  theme_minimal()

barplot(cor_vec, las = 2, col = "steelblue", main = "Correlation with Ctnnal1 Adult")


FeatureScatter(datafil, feature1 = "Ctnnal1", feature2 = "Hhex")
```

```{r}
Idents(so) <- "orig.ident"
# Initialize a data frame to store results
zero_pct <- data.frame(Gene = character(), Identity = character(), PercentZero = numeric())

# Loop through each gene and identity
for (gene in goi) {
  for (id in levels(so$orig.ident)) {
    cells <- WhichCells(so, idents = id)
    expr <- FetchData(so, vars = gene)[cells, gene]
    pct_zero <- sum(expr == 0) / length(expr) * 100
    zero_pct <- rbind(zero_pct, data.frame(Gene = gene, Identity = id, PercentZero = pct_zero))
  }
}

zero_pct$Identity <- factor(zero_pct$Identity, levels = c("HSC.E16.5", "HSC.P7", "HSC.P14", "HSC.Adult"))

# View the result

# Plot
ggplot(zero_pct, aes(x = Identity, y = PercentZero, fill = Gene)) +
 geom_bar(stat = "identity", position = position_dodge()) +
 labs(title = "Percentage of Cells with Zero Expression",
      x = "Sample Identity",
      y = "Percent Zero Expression") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

```


```{r}
vlnage <- VlnPlot(so, features = c("Ctnnal1"), group.by = "orig.ident")
vlnage
ggsave("hscctnnal1vlnplotage.png", vlnage, height = 10, width = 8, dpi = 600)

RidgePlot(so, features = c("Ctnnal1"), group.by = "orig.ident")

```

```{r}
x <- FetchData(so, vars = c("Ctnnal1"))
so$acat <- ifelse(x$Ctnnal1 > 0, TRUE, FALSE)
table(so$acat)

aCat <- subset(so, subset = Ctnnal1 > 0)
Idents(aCat) <- "orig.ident"
degtest <- FindMarkers(aCat, ident.1 = "HPC.E16.5", ident.2 = "HPC.Adult")

```

```{r GFP+ signature}
gfpsig <- read.csv("C:/Users/rohit/OneDrive - Loyola University Chicago/Zhang Lab/RNASeq/Gene Signatures/gfpposoverallsignature.csv")
gfpsig <- gfpsig[,-1]

gfppn <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/RNASeq/Gene Signatures/gfppospnsignature.csv")
gfppn <- gfppn[,-1]

Idents(so) <- "seurat_clusters"

so <- AddModuleScore(so, features = list(gfpsig), name = "GFPpos")
FeaturePlot(so, features = "GFPpos1")
FeaturePlot(so, features = "GFPpos1", split.by = "orig.ident")
VlnPlot(so, features = "GFPpos1")
VlnPlot(so, features = "GFPpos1", split.by = "orig.ident")
VlnPlot(so, features = "GFPpos1", group.by = "orig.ident")


so <- AddModuleScore(so, features = list(gfppn), name = "GFPposPN")
FeaturePlot(so, features = "GFPposPN1")
FeaturePlot(so, features = "GFPposPN1", split.by = "orig.ident")
VlnPlot(so, features = "GFPposPN1")
VlnPlot(so, features = "GFPposPN1", split.by = "orig.ident")
VlnPlot(so, features = "GFPposPN1", group.by = "orig.ident")
Idents(so) <- "acat"
gfpvln <- VlnPlot(so, features = "GFPposPN1", pt.size = 0) + 
  geom_jitter(width=0.1, size=0.5)
gfpvln
gfpvlnsplit <- VlnPlot(so, features = "GFPposPN1", split.by = "orig.ident")
gfpvlnsplit
ggsave("gfpposvlnage.png", gfpvlnsplit, height = 10, width = 8, dpi = 600)
ggsave("gfpposvln.png", gfpvln, height = 10, width = 8, dpi = 600)

ggplot(so, aes(x=so$GFPpos1)) + 
  geom_histogram()

gfpmodule <- so$GFPpos1
gfpmoduledf <- data.frame(gfpmodule)
ggplot(gfpmoduledf, aes(x=gfpmodule)) + 
  geom_histogram()

```

```{r acat stats exploration}
E16 <- subset(so, orig.ident == "HSC.E16.5")
P7 <- subset(so, orig.ident == "HSC.P7")
P14 <- subset(so, orig.ident == "HSC.P14")
Adult <- subset(so, orig.ident == "HSC.Adult")

gfpe16 <- median(E16$GFPpos1)
gfpp7 <- median(P7$GFPpos1)
gfpp14 <- median(P14$GFPpos1)
gfpadult <- median(Adult$GFPpos1)

mediansignature <- data.frame(E16 = gfpe16,
                              P7 = gfpp7, 
                              P14 = gfpp14, 
                              Adult = gfpadult)

gfpe16 <- median(E16$GFPposPN1)
gfpp7 <- median(P7$GFPposPN1)
gfpp14 <- median(P14$GFPposPN1)
gfpadult <- median(Adult$GFPposPN1)

mediansignaturepn <- data.frame(E16 = gfpe16,
                              P7 = gfpp7, 
                              P14 = gfpp14, 
                              Adult = gfpadult)
acattable <- data.frame(E16 = sum(E16$acat)/length(E16$acat),
                        P7 = sum(P7$acat)/length(P7$acat),
                        P14 = sum(P14$acat)/length(P14$acat),
                        Adult = sum(Adult$acat)/length(Adult$acat))

acatlong <- data.frame(
  Population = c("E16", "P7", "P14", "Adult"),
  Proportion = c(sum(E16$acat)/length(E16$acat),
                 sum(P7$acat)/length(P7$acat),
                 sum(P14$acat)/length(P14$acat),
                 sum(Adult$acat)/length(Adult$acat)))

write.csv(acattable, "percentctnnal1pos.csv")

E16acat <- subset(E16, acat == TRUE)
P7acat <- subset(P7, acat == TRUE)
P14acat <- subset(P14, acat == TRUE)
Adultacat <- subset(Adult, acat == TRUE)

ctnnal <- subset(so, acat == TRUE)
x1 <- FetchData(E16acat, vars = c("Ctnnal1"))
x2 <- FetchData(P7acat, vars = c("Ctnnal1"))
x3 <- FetchData(P14acat, vars = c("Ctnnal1"))
x4 <- FetchData(Adultacat, vars = c("Ctnnal1"))
x1$Population <- "E16"
x2$Population <- "P7"
x3$Population <- "P14"
x4$Population <- "Adult"
xdf <- data.frame(E16 = x1$Ctnnal1,
                  P7 = x2$Ctnnal1, 
                  P14 = x3$Ctnnal1,
                  Adult = x4$Ctnnal1)
xdf <- rbind(x1, x2, x3, x4)

xdf$Population <- factor(xdf$Population, levels = c("E16", "P7", "P14", "Adult"))
boxplot <- ggplot(xdf, aes(x = Population, y = Ctnnal1, fill = Population)) + 
  geom_boxplot()
boxplot
ggsave("ctnnal1boxplot.png", boxplot, height = 10, width = 8, dpi = 600)

acatlong$Population <- factor(acatlong$Population, levels = c("E16", "P7", "P14", "Adult"))
ggplot(acatlong, aes(x=Population, y=Proportion, fill = Population)) + 
  geom_col() + 
  labs(title = "Ctnnal1 Positive Cells") + 
  theme_minimal() + 
  theme(plot.title = element_text(hjust = 0.5),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())

```

```{r Heatmap}
negative <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/RNASeq/Gene Signatures/youngnegativedeg.csv")
negative <- negative[,-1]
genelist <- c(gfppn, negative)

write.csv(genelist, "singlecellheatmapgenelist.csv")

fetal <- read.csv("Fetal Signature.csv")

so <- SCTransform(so, assay = "RNA", new.assay.name = "SCT", variable.features.n = nrow(so), verbose = TRUE)
so$acatage <- paste(so$acat, so$orig.ident, sep = "_")
so$acatage <- factor(so$acatage, levels = c("FALSE_HSC.E16.5", "TRUE_HSC.E16.5", "FALSE_HSC.P7", "TRUE_HSC.P7", "FALSE_HSC.P14", "TRUE_HSC.P14", "FALSE_HSC.Adult", "TRUE_HSC.Adult"))


# Create a heatmap

heatmap <- DoHeatmap(
  object = so,
  assay = "SCT",
  features = fetal$X,
  group.by = "acatage"
) + scale_fill_gradientn(colors = c("blue", "white", "red"), values = c(0, 0.5, 1),
  rescaler = function(x, to = c(0, 1), from = range(x, na.rm = TRUE, finite = TRUE)) {
    scales::rescale_mid(x, mid = 0, to = to, from = from)
  })

heatmap

ggsave("hscheatmapfetal.png", heatmap, height = 15, width = 10, dpi = 600)

```

```{r QC}
missing <- c("Scin", "Adcyap1r1", "Acmsd", "Myo18b", "Gm44154", "X1110032F04Rik", "Gm44954", "Zfp175", "Gm38825", "Fam241b", "Sertm2", "X5830416I19Rik", "Gucy2e", "C430002N11Rik", "Gpr15", "Pax9", "Mexis", "Igkv6.32", "Ighv1.80", "Trem6l", "Tpsb2", "Gm3360", "Adra2a", "Ccbe1", "Chsy3", "Mtss2", "Tmem217rt", "Lhfpl6")

missing2 <- c("Scin", "Itgax", "Rnf39", "Adcyap1r1", "Acmsd", "Myo18b", "Gm44154", "X1110032F04Rik", "Gm44954", "Zfp175", "Gm38825", "Rab7b", "Fam241b", "Sertm2", "Ednra", "X5830416I19Rik", "Gucy2e", "C430002N11Rik", "Gpr15", "Pax9", "Mexis", "Igkv6.32", "Ighv1.80", "Trem6l", "Tpsb2", "Gm3360", "Adra2a", "Ccbe1", "Chsy3", "Mtss2", "Tmem217rt", "Lhfpl6")
#there are 32 total missing genes, 6 from the positive and 26 from the negative

posmissing <- Reduce(intersect, list(missing2, gfppn))
negmissing <- Reduce(intersect, list(missing2, negative))



```