---
title: "Tet2 Ripk3 RNA Sequencing Work"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Initial Setup:

Load required packages and set working directory.

```{r packages}
library(DESeq2)
library(tximport)
library(tidyverse)
library(rnaseqGene)
library(pheatmap)
library(ggalt)
library(ggplot2)
```

# DESeq2 analysis
Load data into DESeq dataset

```{r load data}
cts2 <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/raw_counts modified corrected triplicates.csv") #file path to Counts csv document
cts2 <- cts2[, -1] #only necessary if bulk seq data has ensembl id before gene names
cts <- cts2[, -1]
table(duplicated(cts$Gene.name)) #check for duplicate gene names
rownames(cts) <- make.names(cts2$Gene.name, unique = TRUE) #forces gene names to become unique
rm(cts2)
cd <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/coldata.csv") #make coldata file with sample names in one column and then condition in second column
dds <- DESeqDataSetFromMatrix(countData = cts, colData = cd, design = ~ condition)
```

Differential Expression

```{r DESeq}
dds <- DESeq(dds)
rescmp <- results(dds) 
vst <- vst(dds, blind=FALSE)
rlog <- rlog(dds, blind=FALSE)
```

write results as csv

```{r save results}
write.csv(rescmp, "C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/completeresults.csv")
```

### Comparison 1 of 3: DKO-WT

```{r}
res1 <- DESeq2::results(dds, 
                       contrast = c("condition", "DKO", "WT"),
                       independentFiltering = TRUE,
)
write.csv(res1, "C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/dko_wt.csv")
```

### Comparison 2 of 3: Tet2KO-WT

```{r}
res2 <- DESeq2::results(dds, 
                       contrast = c("condition", "Tet2KO", "WT"),
                       independentFiltering = TRUE,
)
write.csv(res2, "C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/tet2_wt.csv")
```

### Comparison 3 of 3: DKO-Tet2KO

```{r}
res3 <- DESeq2::results(dds, 
                       contrast = c("condition", "DKO", "Tet2KO"),
                       independentFiltering = TRUE,
)
write.csv(res3, "C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/dko_tet2ko.csv")
```

#### Filter for p-value

```{r filter for p-value}
res_filtered1 <- res1[which(res1$padj < 0.1 & !is.na(res1$padj)), ]

write.csv(res_filtered1, "C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/dko_wt_filter.csv")

res_filtered2 <- res2[which(res2$padj < 0.1 & !is.na(res1$padj)), ]

write.csv(res_filtered2, "C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/tet2_wt_filter.csv")

res_filtered3 <- res3[which(res3$padj < 0.1 & !is.na(res1$padj)), ]

write.csv(res_filtered3, "C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/dko_tet2ko_filter.csv")
```

display top 20 as heatmap

```{r top 20 heatmap}
select <- order(rowMeans(abs(assay(rlog))),
                decreasing=TRUE)[1:50]
df <- as.data.frame(colData(rlog)[,c("condition")])
color_palette <- colorRampPalette(c("blue", "white", "red"))(100)
pheatmap(assay(rlog)[select,], cluster_rows=FALSE, show_rownames=TRUE,
         cluster_cols=FALSE, color=color_palette, main="Top 20 Heatmap")
```

Attempt at top 20 from each condition heatmap
```{r}

# Assuming you have a DESeqDataSet object called 'dds'
# Perform differential expression analysis for each condition

# Rank genes based on adjusted p-value
top_genes_condition1 <- head(order(res_filtered1$log2FoldChange), 20)
top_genes_condition2 <- head(order(res_filtered2$log2FoldChange), 20)
top_genes_condition3 <- head(order(res_filtered3$log2FoldChange), 20)

# Extract top 20 genes for each condition
top_genes_condition1 <- rownames(res_filtered1)[top_genes_condition1]
top_genes_condition2 <- rownames(res_filtered2)[top_genes_condition2]
top_genes_condition3 <- rownames(res_filtered3)[top_genes_condition3]

# Combine results
top_genes <- unique(c(top_genes_condition1, top_genes_condition2, top_genes_condition3))

# Subset normalized counts to include only the genes of interest
subset_rlog <- assay(rlog)[rownames(rlog) %in% top_genes, ]
subset_rlog <- as.matrix(subset_rlog)

# Define the desired column order
desired_order <- c("WT_1", "WT_2", "WT_3", "Tet2KO_1", "Tet2KO_2", "Tet2KO_3", "DKO_1", "DKO_2", "DKO_3")

# Reorder the columns
subset_rlog <- subset_rlog[, desired_order]

# Extract column data (sample information) and reorder it
col_data <- colData(dds)[desired_order, ]

# Define color palette
color_palette <- colorRampPalette(c("blue", "white", "red"))(100)

# Create a heatmap
heatmap <- pheatmap(subset_rlog, 
         scale = "row", 
         clustering_distance_rows = "euclidean", 
         cluster_cols = FALSE, 
         clustering_method = "complete", 
         color = color_palette,
         show_rownames = TRUE, 
         show_colnames = TRUE,
         main = "Heatmap of AITL Genes",
         treeheight_row = 0)
```
row_order <- order(rowMeans(subset_rlog), decreasing = TRUE)
subset_rlog <- subset_rlog[row_order, ]

## Display selected Genes as Heatmap

```{r}
# List of specific genes
genes_of_interest <- c("Rgs1", "Sh2d1a", "Crip1", "S100a13", "Gabbr1", "Dtnb", "Rsad2", "Cxcl9", 
                       "Kenj10", "Gstm5", "C1qb", "Slc40a1", "Slc43a1", "C1qc", "Veam1", "Marchf1", 
                       "Pla2g2d", "C1qa", "Slc25a37", "Ckap4", "Rgl1", "Mafb", "Pou2af1", "Ighm", 
                       "Il18", "Slco2b1", "Marcks", "Itga9", "Tns3", "Abcc3", "Wdfy3", "Arhgap23")

# Subset normalized counts to include only the genes of interest
subset_rlog <- assay(rlog)[rownames(rlog) %in% genes_of_interest, ]
row_order <- order(rowMeans(subset_rlog), decreasing = TRUE)
subset_rlog <- subset_rlog[row_order, ]
subset_rlog <- as.matrix(subset_rlog)

# Define the desired column order
desired_order <- c("WT_1", "WT_2", "WT_3", "Tet2KO_1", "Tet2KO_2", "Tet2KO_3", "DKO_1", "DKO_2", "DKO_3")

# Reorder the columns
subset_rlog <- subset_rlog[, desired_order]

# Extract column data (sample information) and reorder it
col_data <- colData(dds)[desired_order, ]

# Define color palette
color_palette <- colorRampPalette(c("blue", "white", "red"))(100)

# Create a heatmap
heatmap <- pheatmap(subset_rlog, 
         scale = "row", 
         clustering_distance_rows = "euclidean", 
         cluster_cols = FALSE, 
         clustering_method = "complete", 
         color = color_palette,
         show_rownames = TRUE, 
         show_colnames = TRUE,
         main = "Heatmap of AITL Genes",
         treeheight_row = 0)
```


```{r pca plot}
pca <- plotPCA(rlog)
pca + geom_point(size=5) + 
  theme(
    panel.background = element_rect(fill = "white", color = "black"), # White background
    axis.text = element_blank(), # Remove axis numbers
    axis.ticks = element_blank(), # Remove axis ticks
  ) + 
  scale_color_manual(values = c("WT" = "black", "Tet2KO" = "blue", "DKO" = "red"))
```

##Display GSEA Results

```{r}
mh <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/dkoenriched_mh.csv")
mh <- 
mhplot <- ggplot(mh, aes(x=FDR.q.val, y=reorder(NAME, NES), fill = NES))
mhplot + geom_point(size=3, shape=21) + 
  geom_vline(xintercept = 0.1, linetype = "dashed", color = "red") + 
  theme(
    panel.background = element_rect(fill = "white", color = "black"), 
    plot.title = element_text(hjust = 0.5)
    ) + 
  scale_fill_gradient2(low="blue", mid="white", high="red", midpoint=0, limits=c(-2.5,2.5)) + 
  labs(x="FDR q-value", y="Hallmark Pathways", title="Normalized Enrichment Score")
```

```{r}
sel <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/dkotet2selected.csv")
gseaplot <- ggplot(sel, aes(x=-log(FDR.q.val), y=reorder(NAME, NES), fill = NES))
gseaplot + geom_point(size=5, shape=21) + 
  geom_vline(xintercept = 2, linetype = "dashed", color = "red") + 
  theme(
    panel.background = element_rect(fill = "white", color = "black"), 
    plot.title = element_text(hjust = 0.5)
    ) + 
  scale_fill_gradient2(low="blue", mid="white", high="red", midpoint=0, limits=c(-2.5,2.5)) + 
  labs(x="-log(FDR q-value)", y="Selected Pathways", title="Normalized Enrichment Score")
```

