---
title: "Tet2 Ripk3 RNA Sequencing Analysis"
output:
  html_document: default
  pdf_document: default
editor_options:
  chunk_output_type: console
---

# Initial Setup:

Load required packages (this is assuming you already have these packages installed)

```{r packages, message=FALSE, warning=FALSE}
library(DESeq2)
library(tximport)
library(tidyverse)
library(rnaseqGene)
library(pheatmap)
library(ggalt)
library(cowplot)
```

# DESeq2 analysis

## Load counts matrix into a DESeqDataSet object

```{r Load Data, warning=FALSE, message=FALSE}
cts2 <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/raw_counts modified corrected triplicates.csv") #file path to Counts csv document
cts2 <- cts2[, -1] #only necessary if bulk seq data has ensembl id before gene names
cts <- cts2[, -1]
table(duplicated(cts$Gene.name)) #check for duplicate gene names
rownames(cts) <- make.names(cts2$Gene.name, unique = TRUE) #forces gene names to become unique
rm(cts2)
cd <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/coldata.csv") #make coldata file with sample names in one column and then condition in second column
dds <- DESeqDataSetFromMatrix(countData = cts, colData = cd, design = ~ condition)
```

## Differential Expression analysis

```{r DESeq}
rlog <- rlog(dds, blind=FALSE)
dds <- DESeq(dds)
rescmp <- results(dds) 
```

### Save results as csv

```{r Save Results, eval=FALSE}
write.csv(rescmp, "C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/completeresults.csv")
```

# Generate Heatmap

This heatmap is generated using a list of AITL genes curated using other datasets. 

```{r Heatmap of Specific Genes}
# List of specific genes
genes_of_interest <- c("Rgs1", "Sh2d1a", "Crip1", "S100a13", "Gabbr1", "Dtnb", "Rsad2", "Cxcl9", 
                       "Kenj10", "Gstm5", "C1qb", "Slc40a1", "Slc43a1", "C1qc", "Veam1", "Marchf1", 
                       "Pla2g2d", "C1qa", "Slc25a37", "Ckap4", "Rgl1", "Mafb", "Pou2af1", "Ighm", 
                       "Il18", "Slco2b1", "Marcks", "Itga9", "Tns3", "Abcc3", "Wdfy3", "Arhgap23")

# Subset normalized counts to include only the genes of interest
subset_rlog <- assay(rlog)[rownames(rlog) %in% genes_of_interest, ]
row_order <- order(rowMeans(subset_rlog), decreasing = TRUE)
subset_rlog <- subset_rlog[row_order, ]
subset_rlog <- as.matrix(subset_rlog)

# Define the desired column order
desired_order <- c("WT_1", "WT_2", "WT_3", "Tet2KO_1", "Tet2KO_2", "Tet2KO_3", "DKO_1", "DKO_2", "DKO_3")

# Reorder the columns
subset_rlog <- subset_rlog[, desired_order]

# Extract column data (sample information) and reorder it
col_data <- colData(dds)[desired_order, ]

# Define color palette
color_palette <- colorRampPalette(c("blue", "white", "red"))(100)

# Create a heatmap
heatmap <- pheatmap(subset_rlog, 
         scale = "row", 
         clustering_distance_rows = "euclidean", 
         cluster_cols = FALSE, 
         clustering_method = "complete", 
         color = color_palette,
         show_rownames = TRUE, 
         show_colnames = TRUE,
         main = "Heatmap of AITL Genes",
         treeheight_row = 0)
```

# Generate PCA Plot

```{r PCA Plot}
pca <- plotPCA(rlog)
pca + geom_point(size=5) + 
  theme(
    panel.background = element_rect(fill = "white", color = "black"), # White background
    axis.text = element_blank(),
    axis.ticks = element_blank(),
  ) + 
  scale_color_manual(labels=c("Tet2-/- Ripk3-/-", "Tet2-/-", "WT"), values = c("red","blue", "green3")) + 
  lims(x=c(-15, 25), y=c(-15, 25))
```

# Display GSEA Results

## Filter GSEA reports for an FDR q-value < 0.25 and Normalized Enrichment Score > 1.3

The negative reports were not added for the DKO vs WT comparison as the report was empty, and that created some errors. 
```{r GSEA Results}
dt1 <- read_tsv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/output/20240930/total.Gsea.1727713030884/gsea_report_for_DKO_1727713030884.tsv")
dt2 <- read_tsv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/output/20240930/total.Gsea.1727713030884/gsea_report_for_Tet2KO_1727713030884.tsv")
dt <- bind_rows(dt1, dt2)
rm(dt1, dt2)
dt <- dt[which(dt["FDR q-val"]<0.25 & abs(dt["NES"])>1.3),]
dt <- rename(dt, FDR.q.val="FDR q-val")

write.csv(dt, "C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/output/20240930/filtereddkotet2.csv")

dw <- read_tsv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/output/20240930/totalwt.Gsea.1727713054043/gsea_report_for_DKO_1727713054043.tsv")
dw <- dw[which(dw["FDR q-val"]<0.25 & abs(dw["NES"])>1.3),]
dw <- rename(dw, FDR.q.val="FDR q-val")

write.csv(dw, "C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/output/20240930/filtereddkowt.csv")
```

## Generate Plots

The filtered reports from above were then separated into three categories to then be plotted together. 

#### DKO vs Tet2KO comparison results
```{r Generate Plots - DKO vs Tet2KO, fig.width=10, fig.height=9}
tf <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/output/20240930/total.Gsea.1727713030884/finaltfoll.csv")
tf <- tf[which(tf["FDR.q.val"]<0.25 & abs(tf["NES"])>1.3),]
tfplot <- ggplot(tf, aes(x=-log10(FDR.q.val+0.00001), y=reorder(NAME, -log10(FDR.q.val)), fill = NES))
tfplot + geom_point(size=5, shape=21) + 
 theme(
    panel.background = element_rect(fill = "white"), 
    panel.border = element_rect(color = "black", fill = NA), 
    plot.title = element_text(hjust = 0.5), 
    axis.text=element_text(color="black", face="bold")
    ) +
  scale_fill_gradient2(low="blue", mid="white", high="red", midpoint=0, limits=c(-2.4,2.4)) + 
  labs(x="-log(FDR q value)", y="Genesets", title="T cell & Lymphoma Genesets") +
  xlim(0,5.1)

cc <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/output/20240930/total.Gsea.1727713030884/finalcellcycle.csv")
cc <- cc[which(cc["FDR.q.val"]<0.25 & abs(cc["NES"])>1.3),]
ccplot <- ggplot(cc, aes(x=-log10(FDR.q.val+0.00001), y=reorder(NAME, -log10(FDR.q.val)), fill = NES))
ccplot + geom_point(size=5, shape=21) +  
 theme(
    panel.background = element_rect(fill = "white"), 
    panel.border = element_rect(color = "black", fill = NA), 
    plot.title = element_text(hjust = 0.5), 
    axis.text=element_text(color="black", face="bold")
    ) +
  scale_fill_gradient2(low="blue", mid="white", high="red", midpoint=0, limits=c(-2.2,2.2)) + 
  labs(x="-log(FDR q-value)", y="Genesets", title="Cell Cycle & Proliferation Genesets") +
  xlim(0,5.1)

my <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/output/20240930/total.Gsea.1727713030884/finalmyeloid.csv")
my <- my[which(my["FDR.q.val"]<0.25 & abs(my["NES"])>1.3),]
myplot <- ggplot(my, aes(x=-log10(FDR.q.val+0.00001), y=reorder(NAME, -log10(FDR.q.val)), fill = NES))
myplot + geom_point(size=5, shape=21) +  
 theme(
    panel.background = element_rect(fill = "white"), 
    panel.border = element_rect(color = "black", fill = NA), 
    plot.title = element_text(hjust = 0.5), 
    axis.text=element_text(color="black", face="bold")
    ) +
  scale_fill_gradient2(low="blue", mid="white", high="red", midpoint=0, limits=c(-2.0,2.0)) + 
  labs(x="-log(FDR q-value)", y="Genesets", title="Myeloid Genesets") +
  xlim(0,5.1)
```

#### DKO vs WT comparison results
```{r Generate Plots - DKO vs WT, fig.width=10, fig.height=9}
tf <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/output/20240930/totalwt.Gsea.1727713054043/finaltfoll.csv")
tf <- tf[which(tf["FDR.q.val"]<0.25 & abs(tf["NES"])>1.3),]
tfplot <- ggplot(tf, aes(x=-log10(FDR.q.val+0.00001), y=reorder(NAME, -log10(FDR.q.val)), fill = NES))
tfplot + geom_point(size=5, shape=21) + 
 theme(
    panel.background = element_rect(fill = "white"), 
    panel.border = element_rect(color = "black", fill = NA), 
    plot.title = element_text(hjust = 0.5), 
    axis.text=element_text(color="black", face="bold")
    ) +
  scale_fill_gradient2(low="blue", mid="white", high="red", midpoint=0, limits=c(-1.9,1.9)) + 
  labs(x="-log(FDR q value)", y="Genesets", title="T cell & Lymphoma Genesets") +
  xlim(0,5.1)

cc <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/output/20240930/totalwt.Gsea.1727713054043/finalcellcycle.csv")
cc <- cc[which(cc["FDR.q.val"]<0.25 & abs(cc["NES"])>1.3),]
ccplot <- ggplot(cc, aes(x=-log10(FDR.q.val+0.00001), y=reorder(NAME, -log10(FDR.q.val)), fill = NES))
ccplot + geom_point(size=5, shape=21) +  
 theme(
    panel.background = element_rect(fill = "white"), 
    panel.border = element_rect(color = "black", fill = NA), 
    plot.title = element_text(hjust = 0.5), 
    axis.text=element_text(color="black", face="bold")
    ) +
  scale_fill_gradient2(low="blue", mid="white", high="red", midpoint=0, limits=c(-1.7,1.7)) + 
  labs(x="-log(FDR q-value)", y="Genesets", title="Cell Cycle & Proliferation Genesets") +
  xlim(0,5.1)

my <- read.csv("C:/Users/rthalla/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/output/20240930/totalwt.Gsea.1727713054043/finalmyeloid.csv")
my <- my[which(my["FDR.q.val"]<0.25 & abs(my["NES"])>1.3),]
myplot <- ggplot(my, aes(x=-log10(FDR.q.val+0.00001), y=reorder(NAME, -log10(FDR.q.val)), fill = NES))
myplot + geom_point(size=5, shape=21) +  
 theme(
    panel.background = element_rect(fill = "white"), 
    panel.border = element_rect(color = "black", fill = NA), 
    plot.title = element_text(hjust = 0.5), 
    axis.text=element_text(color="black", face="bold")
    ) +
  scale_fill_gradient2(low="blue", mid="white", high="red", midpoint=0, limits=c(-1.5,1.5)) + 
  labs(x="-log(FDR q-value)", y="Genesets", title="Myeloid Genesets") +
  xlim(0,5.1)
```
