---
title: "Tet2 Ripk3 RNA Sequencing Analysis"
output:
  html_document: default
  pdf_document: default
editor_options:
  chunk_output_type: console
---

# Initial Setup:

Load required packages (this is assuming you already have these packages installed). Always run this first.

```{r packages, message=FALSE, warning=FALSE}
library(tidyverse)
library(DESeq2)
library(rnaseqGene)
library(pheatmap)
library(ggalt)
library(limma)
library(stringr)
library(ggrepel)

setwd("C:/Users/rohit/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/")
```

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = 'C:/Users/rohit/OneDrive - Loyola University Chicago/Zhang Lab/Kanak Tet2 Ripk3/')
```

# DESeq2 analysis

## Load counts matrix into a DESeqDataSet object

```{r Load Data, warning=FALSE, message=FALSE}
cts2 <- read.csv("raw_counts modified corrected triplicates.csv") #file path to Counts csv document
cts2 <- cts2[, -1] #only necessary if bulk seq data has ensembl id before gene names
cts <- cts2[, -1]
table(duplicated(cts$Gene.name)) #check for duplicate gene names
rownames(cts) <- make.names(cts2$Gene.name, unique = TRUE) #forces gene names to become unique
rm(cts2)
cd <- read.csv("coldata.csv") #make coldata file with sample names in one column and then condition in second column
dds <- DESeqDataSetFromMatrix(countData = cts, colData = cd, design = ~ condition)
```

## Differential Expression analysis

```{r DESeq}
rlog <- rlog(dds, blind=FALSE)
dds <- DESeq(dds)
rescmp <- results(dds) 

res_DKO_vs_WT <- results(dds, contrast=c("condition", "DKO", "WT"), alpha = 0.05)
resDKO_WT <- as.data.frame(res_DKO_vs_WT)
```

### Save results as csv

```{r Save Results, eval=FALSE}
write.csv(rescmp, "completeresults.csv")
```

# Generate PCA Plot

```{r PCA Plot}
pca <- plotPCA(rlog)
pca + geom_point(size=5) + 
  theme(
    panel.background = element_rect(fill = "white", color = "black"), # White background
    axis.text = element_blank(),
    axis.ticks = element_blank(),
  ) + 
  scale_color_manual(labels=c("Tet2-/- Ripk3-/-", "Tet2-/-", "WT"), values = c("red","blue", "green3")) + 
  lims(x=c(-15, 25), y=c(-15, 25))
```

# Volcano Plots

## DKO vs WT
```{r volcano plot DKO vs WT}
res_DKO_vs_WT <- results(dds, contrast=c("condition", "DKO", "WT"), alpha = 0.05)
res_DKO_vs_WT <- res_DKO_vs_WT %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene") %>% 
  mutate(
    sig = case_when(
      padj <= 0.05 & log2FoldChange >0  ~ "Up",
      padj <= 0.05 & log2FoldChange <0 ~ "Down",
      TRUE ~ "NS"
    )
  )

top10_up <- res_DKO_vs_WT %>%
  filter(sig == "Up") %>%
  arrange(pvalue) %>%
  head(10)

top10_down <- res_DKO_vs_WT %>%
  filter(sig == "Down") %>%
  arrange(pvalue) %>%
  head(10)

leading_edge <- bind_rows(top10_up, top10_down)

volcano <- ggplot(res_DKO_vs_WT, aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point(aes(color = sig), alpha = 0.8, size = 1.8) +
  scale_color_manual(
    values = c(
      "Up" = "red",
      "Down" = "blue",
      "NS" = "gray70"
    )
  ) +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "black") +
  labs(
    title = expression("Tet2KO Ripk3KO vs WT"),
    x = "log2 Fold Change",
    y = "-log10 Adjusted p-value",
    color = "Direction"
  ) +
  theme_bw(base_size = 12) + 
  geom_text_repel(
    data = leading_edge,
    aes(label = gene), 
    size = 3, 
    max.overlaps = Inf, 
    box.padding = 0.4, 
    point.padding = 0.2, 
    segment.color = "gray50")
volcano

ggsave("volcano_dkowt.png", volcano, dpi = 600)
```

## DKO vs Tet2KO

```{r volcano plot DKO vs Tet2}
res_DKO_vs_Tet2 <- results(dds, contrast=c("condition", "DKO", "Tet2KO"), alpha = 0.05)
res_DKO_vs_Tet2 <- res_DKO_vs_Tet2 %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene") %>% 
  mutate(
    sig = case_when(
      padj <= 0.05 & log2FoldChange >0  ~ "Up",
      padj <= 0.05 & log2FoldChange <0 ~ "Down",
      TRUE ~ "NS"
    )
  )

top10_up <- res_DKO_vs_Tet2 %>%
  filter(sig == "Up") %>%
  arrange(pvalue) %>%
  head(10)

top10_down <- res_DKO_vs_Tet2 %>%
  filter(sig == "Down") %>%
  arrange(pvalue) %>%
  head(10)

leading_edge <- bind_rows(top10_up, top10_down)

volcano <- ggplot(res_DKO_vs_Tet2, aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point(aes(color = sig), alpha = 0.8, size = 1.8) +
  scale_color_manual(
    values = c(
      "Up" = "red",
      "Down" = "blue",
      "NS" = "gray70"
    )
  ) +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "black") +
  labs(
    title = expression("Tet2KO Ripk3KO vs Tet2KO"),
    x = "log2 Fold Change",
    y = "-log10 Adjusted p-value",
    color = "Direction"
  ) +
  theme_bw(base_size = 12) + 
  geom_text_repel(
    data = leading_edge,
    aes(label = gene), 
    size = 3, 
    max.overlaps = Inf, 
    box.padding = 0.4, 
    point.padding = 0.2, 
    segment.color = "gray50")
volcano

ggsave("volcano_dkotet2.png", volcano, dpi = 600)

```

# Creating curated AITL Gene List

#### GSE221599
```{r Setup2}
cts2 <- read.csv("Gene List Validation/GSE221599.csv") 
cts2 <- cts2[, -1]
cts <- cts2[, -1]
table(duplicated(cts$gene_name))
rownames(cts) <- make.names(cts2$gene_name, unique = TRUE) 
rm(cts2)
cd <- read.csv("Gene List Validation/coldataGSE221599.csv") 
dds <- DESeqDataSetFromMatrix(countData = cts, colData = cd, design = ~ condition)
```

```{r Analysis2}
dds <- DESeq(dds)
res_DM_vs_WT <- results(dds, contrast=c("condition", "DM", "WT"), alpha = 0.05)
res221599 <- as.data.frame(res_DM_vs_WT)
write.csv(res221599, "Gene List Validation/res221599v2.csv")
```

#### GSE90976
```{r Setup9}
cts2 <- read.csv("Gene List Validation/GSE90976.csv") 
cts <- cts2[, -1]
table(duplicated(cts$Genes)) #check for duplicate gene names
rownames(cts) <- make.names(cts2$Genes, unique = TRUE)
rm(cts2)
cd <- read.csv("Gene List Validation/coldataGSE90976.csv") 
design <- model.matrix(~0 + condition, data=cd)
cts <- log2(cts+1)
```

```{r Analysis9}
contrast_matrix <- makeContrasts(DKO_vs_WT = conditionDKO - conditionWT, levels = design)
GSE90976 <- lmFit(cts, design)
GSE90976 <- contrasts.fit(GSE90976, contrast_matrix)
GSE90976 <- eBayes(GSE90976)
res90976 <- topTable(GSE90976, coef = "DKO_vs_WT", adjust = "fdr", number = Inf)
write.csv(res90976, "Gene List Validation/results90976.csv")
```

#### Overlap

```{r Overlap}
# Rename padj to adj.P.Val
colnames(res221599)[colnames(res221599) == "padj"] <- "adj.P.Val"
# Optional: Rename log2FoldChange to logFC to match limma exactly
colnames(res221599)[colnames(res221599) == "log2FoldChange"] <- "logFC"

filter_genes <- function(df, direction = "up", p_thresh = 0.1, fc_thresh = 0.5) {
  df$gene <- sub("\\..*", "", rownames(df))
  
  if (direction == "up") {
    subset_df <- df[which(df$logFC > fc_thresh & df$adj.P.Val < p_thresh), ]
  } else {
    subset_df <- df[which(df$logFC < -fc_thresh & df$adj.P.Val < p_thresh), ]
  }
  return(subset_df$gene)
}

up_221599 <- filter_genes(res221599, "up")
up_90976  <- filter_genes(res90976,  "up")

down_221599 <- filter_genes(res221599, "down")
down_90976  <- filter_genes(res90976,  "down")

overlap_up <- Reduce(intersect, list(up_221599, up_90976))
overlap_down <- Reduce(intersect, list(down_221599, down_90976))

final_list <- data.frame(
  gene = c(overlap_up, overlap_down),
  direction = c(rep("Consistent_Up", length(overlap_up)), 
                rep("Consistent_Down", length(overlap_down)))
)

write.csv(final_list, "Gene List Validation/consistent_overlap.csv", row.names = FALSE)
```

# Generate Heatmap

This heatmap is generated using a list of AITL genes curated using other datasets. 

```{r Heatmap of Specific Genes}
rlog_mat <- assay(rlog)
subset_rlog <- rlog_mat[rownames(rlog_mat) %in% final_list$gene, ]
dko_samples <- c("DKO_1", "DKO_2", "DKO_3")
rest <- c("WT_1", "WT_2", "WT_3", "Tet2KO_1", "Tet2KO_2", "Tet2KO_3")
row_order <- order(rowMeans(subset_rlog[, dko_samples])-rowMeans(subset_rlog[, rest]), decreasing = FALSE)
row_order <- c(11, 30, 31, 1, 13, 14, 5, 42, 20, 28, 8, 25, 34, 38, 17, 18, 4, 37, 22, 35, 26, 19, 7, 9, 12, 2, 15, 6, 41, 39, 27, 23, 43, 44, 10, 40, 32, 24, 33, 36, 29, 21, 3, 16)
subset_rlog <- subset_rlog[row_order, ]
subset_rlog <- as.matrix(subset_rlog)

desired_order <- c("WT_1", "WT_2", "WT_3", "Tet2KO_1", "Tet2KO_2", "Tet2KO_3", "DKO_1", "DKO_2", "DKO_3")

subset_rlog <- subset_rlog[, desired_order]

col_data <- colData(dds)[desired_order, ]

color_palette <- colorRampPalette(c("blue", "white", "red"))(100)

heatmap <- pheatmap(subset_rlog,
                    scale = "row",
         color = color_palette,
         show_rownames = TRUE, 
         show_colnames = TRUE,
         main = "Heatmap of AITL Genes",
         treeheight_row = 0,
         cluster_rows = FALSE,
         cluster_cols = FALSE,)
heatmap

ggsave("aitlsigheatmap_originalorder.png", heatmap, dpi = 600, height = 8, width = 6)

```

# Display GSEA Results

## Filter GSEA reports for an FDR q-value < 0.25 and Normalized Enrichment Score > 1.3


```{r GSEA Results tlymph}
dw1 <- read_tsv("output/dkowt_tlymph.GseaPreranked.1769300113793/gsea_report_for_na_pos_1769300113793.tsv")
dw2 <- read_tsv("output/dkowt_tlymph.GseaPreranked.1769300113793/gsea_report_for_na_neg_1769300113793.tsv")
dw <- bind_rows(dw1, dw2)
rm(dw1, dw2)
dw <- dw[which(dw["FDR q-val"]<0.25 & abs(dw["NES"])>1.0),]

write.csv(dw, "output/filtereddkowt_tlymph.csv")

dt1 <- read_tsv("output/dkotet2_tlymph.GseaPreranked.1769300784692/gsea_report_for_na_pos_1769300784692.tsv")
dt2 <- read_tsv("output/dkotet2_tlymph.GseaPreranked.1769300784692/gsea_report_for_na_neg_1769300784692.tsv")
dt <- bind_rows(dt1, dt2)
rm(dt1, dt2)
dt <- dt[which(dt["FDR q-val"]<0.25 & abs(dt["NES"])>1.0),]

write.csv(dt, "output/filtereddkotet2_tlymph.csv")
```

```{r GSEA Results cell cycle}
dw1 <- read_tsv("output/dkowt_cellcycle.GseaPreranked.1769300936240/gsea_report_for_na_pos_1769300936240.tsv")
dw2 <- read_tsv("output/dkowt_cellcycle.GseaPreranked.1769300936240/gsea_report_for_na_neg_1769300936240.tsv")
dw <- bind_rows(dw1, dw2)
rm(dw1, dw2)
dw <- dw[which(dw["FDR q-val"]<0.25 & abs(dw["NES"])>1.0),]

write.csv(dw, "output/filtereddkowt_cellcycle.csv")

dt1 <- read_tsv("output/dkotet2_cellcycle.GseaPreranked.1769300922079/gsea_report_for_na_pos_1769300922079.tsv")
dt2 <- read_tsv("output/dkotet2_cellcycle.GseaPreranked.1769300922079/gsea_report_for_na_neg_1769300922079.tsv")
dt <- bind_rows(dt1, dt2)
rm(dt1, dt2)
dt <- dt[which(dt["FDR q-val"]<0.25 & abs(dt["NES"])>1.0),]

write.csv(dt, "output/filtereddkotet2_cellcycle.csv")
```

## Generate Plots

#### DKO vs Tet2KO comparison results
```{r Generate Plots - DKO vs Tet2KO, fig.width=10, fig.height=9}
tf2 <- read.csv("output/filtereddkotet2_tlymph.csv")
tf2$NAME <- gsub("_", " ", tf2$NAME)
tfplot <- ggplot(tf2, aes(x=-log10(FDR.q.val+0.00001), y=reorder(NAME, NES), fill = NES, size = SIZE)) +
  geom_point(shape=21) + 
  scale_size(range = c(5, 10)) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 30)) +
 theme(
    panel.background = element_rect(fill = "white"), 
    panel.border = element_rect(color = "black", fill = NA), 
    plot.title = element_text(hjust = 0.5), 
    axis.text=element_text(color="black", face="bold")
    ) +
  scale_fill_gradient2(low="blue", mid="white", high="red", midpoint=0, limits=c(-2.5,2.5)) + 
  labs(x="-log(FDR q value)", y="Genesets", title="T cell & Lymphoma Genesets") +
  xlim(0,5.1) + 
  geom_vline(xintercept = -log10(0.25), linetype = "dashed")
tfplot

cc2 <- read.csv("output/filtereddkotet2_cellcycle.csv")
cc2$NAME <- gsub("_", " ", cc2$NAME)
cc2 <- cc2[cc2$NAME %in% cc2$NAME, ]
ccplot <- ggplot(cc2, aes(x=-log10(FDR.q.val+0.00001), y=reorder(NAME, NES), fill = NES, size = SIZE)) +
  geom_point(shape=21) +  
  scale_size(range = c(5, 10)) + 
  scale_y_discrete(labels = function(x) str_wrap(x, width = 30)) +
 theme(
    panel.background = element_rect(fill = "white"), 
    panel.border = element_rect(color = "black", fill = NA), 
    plot.title = element_text(hjust = 0.5), 
    axis.text=element_text(color="black", face="bold")
    ) +
  scale_fill_gradient2(low="blue", mid="white", high="red", midpoint=0, limits=c(-3,3)) + 
  labs(x="-log(FDR q-value)", y="Genesets", title="Cell Cycle & Proliferation Genesets") +
  xlim(0,5.1) + 
  geom_vline(xintercept = -log10(0.25), linetype = "dashed")
ccplot

ggsave("dkotet2_tlymph.png", tfplot, dpi = 600, height = 6.5, width = 6.5)

ggsave("dkotet2_cellcycle.png", ccplot, dpi = 600, height = 6.5, width = 6.5)
```

#### DKO vs WT comparison results
```{r Generate Plots - DKO vs WT, fig.width=10, fig.height=9}
tf <- read.csv("output/filtereddkowt_tlymph.csv")
tf$NAME <- gsub("_", " ", tf$NAME)
tfplot <- ggplot(tf, aes(x=-log10(FDR.q.val+0.00001), y=reorder(NAME, NES), fill = NES, size = SIZE)) +
  geom_point(shape=21) + 
  scale_size(range = c(5, 10)) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 30)) +
 theme(
    panel.background = element_rect(fill = "white"), 
    panel.border = element_rect(color = "black", fill = NA), 
    plot.title = element_text(hjust = 0.5), 
    axis.text=element_text(color="black", face="bold")
    ) +
  scale_fill_gradient2(low="blue", mid="white", high="red", midpoint=0, limits=c(-2.5,2.5)) + 
  labs(x="-log(FDR q value)", y="Genesets", title="T cell & Lymphoma Genesets") +
  xlim(0,5.1) + 
  geom_vline(xintercept = -log10(0.25), linetype = "dashed")
tfplot

cc <- read.csv("output/filtereddkowt_cellcycle.csv")
cc$NAME <- gsub("_", " ", cc$NAME)
ccplot <- ggplot(cc, aes(x=-log10(FDR.q.val+0.00001), y=reorder(NAME, NES), fill = NES, size = SIZE)) +
  geom_point(shape=21) +  
  scale_size(range = c(5, 10)) + 
  scale_y_discrete(labels = function(x) str_wrap(x, width = 30)) +
 theme(
    panel.background = element_rect(fill = "white"), 
    panel.border = element_rect(color = "black", fill = NA), 
    plot.title = element_text(hjust = 0.5), 
    axis.text=element_text(color="black", face="bold")
    ) +
  scale_fill_gradient2(low="blue", mid="white", high="red", midpoint=0, limits=c(-3,3)) + 
  labs(x="-log(FDR q-value)", y="Genesets", title="Cell Cycle & Proliferation Genesets") +
  xlim(0,5.1) + 
  geom_vline(xintercept = -log10(0.25), linetype = "dashed")
ccplot

ggsave("dkowt_tlymph.png", tfplot, dpi = 600, height = 6.5, width = 6.5)

ggsave("dkowt_cellcycle.png", ccplot, dpi = 600, height = 6.5, width = 6.5)
```

# ICOS RNA-seq expression

```{r icos expression}
icos <- rlog_mat["Icos",]
write.csv(icos, "icosexpression.csv")
```